{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Finetuning PubmedBERT for citation intent classification"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "31cb3b21-41b7-4b63-b089-9083d38b61b6",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>id</th>\n",
       "      <th>sentence</th>\n",
       "      <th>used</th>\n",
       "      <th>created</th>\n",
       "      <th>mention</th>\n",
       "      <th>context</th>\n",
       "      <th>label</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>PMC5189946</td>\n",
       "      <td>All of this analysis was implemented using Mat...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>All of this analysis was implemented using Mat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>PMC4511233</td>\n",
       "      <td>Code for calculating partition similarity, obt...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Since the probability of getting a given MI is...</td>\n",
       "      <td>0</td>\n",
       "      <td>Code for calculating partition similarity, obt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>PMC4186879</td>\n",
       "      <td>All behavioral statistical analyses were perfo...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>All behavioral statistical analyses were perfo...</td>\n",
       "      <td>2</td>\n",
       "      <td>All behavioral statistical analyses were perfo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>PMC5026371</td>\n",
       "      <td>M-Track was written using Python 2.7, OpenCV 3...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>M-Track was written using Python 2.7, OpenCV 3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>PMC1283974</td>\n",
       "      <td>Mindboggle is a freely downloadable, open sour...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Mindboggle is a freely downloadable, open sour...</td>\n",
       "      <td>0</td>\n",
       "      <td>Mindboggle is a freely downloadable, open sour...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4183</th>\n",
       "      <td>995</td>\n",
       "      <td>PMC2936424</td>\n",
       "      <td>Study sample\\n</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Study sample\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4184</th>\n",
       "      <td>996</td>\n",
       "      <td>PMC3660501</td>\n",
       "      <td>U0216 and Wortmannin were dissolved in PBS whe...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>U0216 and Wortmannin were dissolved in PBS whe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4185</th>\n",
       "      <td>997</td>\n",
       "      <td>PMC4213368</td>\n",
       "      <td>The linker unit serves as an acceptor for the ...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>The linker unit serves as an acceptor for the ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4186</th>\n",
       "      <td>998</td>\n",
       "      <td>PMC4451992</td>\n",
       "      <td>The Training and Recourse Center of CIDI in Be...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>The Training and Recourse Center of CIDI in Be...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4187</th>\n",
       "      <td>999</td>\n",
       "      <td>PMC5869224</td>\n",
       "      <td>Fluorescent PCR permits multiplexing of differ...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Fluorescent PCR permits multiplexing of differ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4188 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0  ...                                               text\n",
       "0              0  ...  All of this analysis was implemented using Mat...\n",
       "1              1  ...  Code for calculating partition similarity, obt...\n",
       "2              2  ...  All behavioral statistical analyses were perfo...\n",
       "3              3  ...  M-Track was written using Python 2.7, OpenCV 3...\n",
       "4              4  ...  Mindboggle is a freely downloadable, open sour...\n",
       "...          ...  ...                                                ...\n",
       "4183         995  ...                                     Study sample\\n\n",
       "4184         996  ...  U0216 and Wortmannin were dissolved in PBS whe...\n",
       "4185         997  ...  The linker unit serves as an acceptor for the ...\n",
       "4186         998  ...  The Training and Recourse Center of CIDI in Be...\n",
       "4187         999  ...  Fluorescent PCR permits multiplexing of differ...\n",
       "\n",
       "[4188 rows x 9 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from datasets import Dataset\n",
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('../data/software_citation_intent_merged.csv')\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "2959218f-220d-4609-ad56-fe0dff812014",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>All statistical analyses were performed with S...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>The subjects underwent prenatal follow-up, and...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>The data were analyzed with SPSS software (IBM...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>We developed an automated image analysis algor...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Categorizing the free response answers involve...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4183</th>\n",
       "      <td>These studies have not been included in the pr...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4184</th>\n",
       "      <td>Hierarchical Bayesian parameter estimation usi...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4185</th>\n",
       "      <td>°C and tools offered by all providers through ...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4186</th>\n",
       "      <td>It measures land use intensity only of the ind...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4187</th>\n",
       "      <td>To calculate ORs and SE from the ALSPAC BMI su...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4188 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   text  label\n",
       "0     All statistical analyses were performed with S...      1\n",
       "1     The subjects underwent prenatal follow-up, and...      3\n",
       "2     The data were analyzed with SPSS software (IBM...      1\n",
       "3     We developed an automated image analysis algor...      0\n",
       "4     Categorizing the free response answers involve...      3\n",
       "...                                                 ...    ...\n",
       "4183  These studies have not been included in the pr...      3\n",
       "4184  Hierarchical Bayesian parameter estimation usi...      1\n",
       "4185  °C and tools offered by all providers through ...      2\n",
       "4186  It measures land use intensity only of the ind...      3\n",
       "4187  To calculate ORs and SE from the ALSPAC BMI su...      1\n",
       "\n",
       "[4188 rows x 2 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset = df[['text', 'label']]\n",
    "dataset = dataset.sample(n=len(dataset), random_state=42)\n",
    "dataset = dataset.reset_index(drop=True)\n",
    "dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "c959444b-1dff-4b16-b505-d131dde484e4",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "id2label = {0: \"created\", 1: \"used\", 2: \"mention\", 3: \"none\"}\n",
    "label2id = {\"created\": 0, \"used\": 1, \"mention\": 2, \"none\": 3}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "ae2de3d0-6380-42c0-88a0-30cf0280d3a7",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Dataset({\n",
       "    features: ['text', 'label'],\n",
       "    num_rows: 4188\n",
       "})"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset = Dataset.from_pandas(dataset)\n",
    "dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "e809c74e-a8d6-40c2-8c53-966d966d4e69",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "datasets.arrow_dataset.Dataset"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(dataset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "4bbd64f8-23a4-4780-a6bf-bdac966dfb79",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "dataset = dataset.train_test_split(test_size=0.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "59797e50-bf1a-4c43-be8b-e1840c88b76b",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "datasets.dataset_dict.DatasetDict"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(dataset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "d284d73a-a57d-436a-8758-692873d68336",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "838"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dataset['test'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "7fa85b40-8e2b-4d30-a200-e59a18e33284",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'text': 'Thus, an additional MRI model has been included in the DCE@urLAB application and is discussed later.\\n',\n",
       " 'label': 0}"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset['train'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "ab9b4cbe-dcd0-465b-8916-e19307054771",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2023-10-27 16:24:06,491] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/databricks/python/lib/python3.10/site-packages/transformers/generation_utils.py:24: FutureWarning: Importing `GenerationMixin` from `src/transformers/generation_utils.py` is deprecated and will be removed in Transformers v5. Import as `from transformers import GenerationMixin` instead.\n",
      "  warnings.warn(\n",
      "/databricks/python/lib/python3.10/site-packages/transformers/generation_tf_utils.py:24: FutureWarning: Importing `TFGenerationMixin` from `src/transformers/generation_tf_utils.py` is deprecated and will be removed in Transformers v5. Import as `from transformers import TFGenerationMixin` instead.\n",
      "  warnings.warn(\n",
      "Loading custom CUDA kernels...\n",
      "Loading custom CUDA kernels...\n",
      "Using /root/.cache/torch_extensions/py310_cu118 as PyTorch extensions root...\n",
      "Creating extension directory /root/.cache/torch_extensions/py310_cu118/cuda_kernel...\n",
      "Detected CUDA files, patching ldflags\n",
      "Emitting ninja build file /root/.cache/torch_extensions/py310_cu118/cuda_kernel/build.ninja...\n",
      "Building extension module cuda_kernel...\n",
      "Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)\n",
      "Loading extension module cuda_kernel...\n",
      "Xformers is not installed correctly. If you want to use memory_efficient_attention to accelerate training use the following command to install Xformers\n",
      "pip install xformers.\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e7085a3c4e6b4ee998665ba2a0ab7091",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading (…)okenizer_config.json:   0%|          | 0.00/28.0 [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d4ac1d3506f24794b30f06c374ea6d24",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading (…)lve/main/config.json:   0%|          | 0.00/385 [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "loading configuration file config.json from cache at /root/.cache/huggingface/hub/models--microsoft--BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext/snapshots/eaa409b6b7c9380a5f2ba7a59aa97712ff30f386/config.json\n",
      "Model config BertConfig {\n",
      "  \"_name_or_path\": \"microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext\",\n",
      "  \"architectures\": [\n",
      "    \"BertForMaskedLM\"\n",
      "  ],\n",
      "  \"attention_probs_dropout_prob\": 0.1,\n",
      "  \"classifier_dropout\": null,\n",
      "  \"hidden_act\": \"gelu\",\n",
      "  \"hidden_dropout_prob\": 0.1,\n",
      "  \"hidden_size\": 768,\n",
      "  \"initializer_range\": 0.02,\n",
      "  \"intermediate_size\": 3072,\n",
      "  \"layer_norm_eps\": 1e-12,\n",
      "  \"max_position_embeddings\": 512,\n",
      "  \"model_type\": \"bert\",\n",
      "  \"num_attention_heads\": 12,\n",
      "  \"num_hidden_layers\": 12,\n",
      "  \"pad_token_id\": 0,\n",
      "  \"position_embedding_type\": \"absolute\",\n",
      "  \"transformers_version\": \"4.31.0\",\n",
      "  \"type_vocab_size\": 2,\n",
      "  \"use_cache\": true,\n",
      "  \"vocab_size\": 30522\n",
      "}\n",
      "\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "01cb536ff03340bbb8f23005775ca63f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading (…)solve/main/vocab.txt:   0%|          | 0.00/226k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "loading file vocab.txt from cache at /root/.cache/huggingface/hub/models--microsoft--BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext/snapshots/eaa409b6b7c9380a5f2ba7a59aa97712ff30f386/vocab.txt\n",
      "loading file tokenizer.json from cache at None\n",
      "loading file added_tokens.json from cache at None\n",
      "loading file special_tokens_map.json from cache at None\n",
      "loading file tokenizer_config.json from cache at /root/.cache/huggingface/hub/models--microsoft--BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext/snapshots/eaa409b6b7c9380a5f2ba7a59aa97712ff30f386/tokenizer_config.json\n",
      "loading configuration file config.json from cache at /root/.cache/huggingface/hub/models--microsoft--BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext/snapshots/eaa409b6b7c9380a5f2ba7a59aa97712ff30f386/config.json\n",
      "Model config BertConfig {\n",
      "  \"_name_or_path\": \"microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext\",\n",
      "  \"architectures\": [\n",
      "    \"BertForMaskedLM\"\n",
      "  ],\n",
      "  \"attention_probs_dropout_prob\": 0.1,\n",
      "  \"classifier_dropout\": null,\n",
      "  \"hidden_act\": \"gelu\",\n",
      "  \"hidden_dropout_prob\": 0.1,\n",
      "  \"hidden_size\": 768,\n",
      "  \"initializer_range\": 0.02,\n",
      "  \"intermediate_size\": 3072,\n",
      "  \"layer_norm_eps\": 1e-12,\n",
      "  \"max_position_embeddings\": 512,\n",
      "  \"model_type\": \"bert\",\n",
      "  \"num_attention_heads\": 12,\n",
      "  \"num_hidden_layers\": 12,\n",
      "  \"pad_token_id\": 0,\n",
      "  \"position_embedding_type\": \"absolute\",\n",
      "  \"transformers_version\": \"4.31.0\",\n",
      "  \"type_vocab_size\": 2,\n",
      "  \"use_cache\": true,\n",
      "  \"vocab_size\": 30522\n",
      "}\n",
      "\n",
      "loading configuration file config.json from cache at /root/.cache/huggingface/hub/models--microsoft--BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext/snapshots/eaa409b6b7c9380a5f2ba7a59aa97712ff30f386/config.json\n",
      "Model config BertConfig {\n",
      "  \"_name_or_path\": \"microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext\",\n",
      "  \"architectures\": [\n",
      "    \"BertForMaskedLM\"\n",
      "  ],\n",
      "  \"attention_probs_dropout_prob\": 0.1,\n",
      "  \"classifier_dropout\": null,\n",
      "  \"hidden_act\": \"gelu\",\n",
      "  \"hidden_dropout_prob\": 0.1,\n",
      "  \"hidden_size\": 768,\n",
      "  \"initializer_range\": 0.02,\n",
      "  \"intermediate_size\": 3072,\n",
      "  \"layer_norm_eps\": 1e-12,\n",
      "  \"max_position_embeddings\": 512,\n",
      "  \"model_type\": \"bert\",\n",
      "  \"num_attention_heads\": 12,\n",
      "  \"num_hidden_layers\": 12,\n",
      "  \"pad_token_id\": 0,\n",
      "  \"position_embedding_type\": \"absolute\",\n",
      "  \"transformers_version\": \"4.31.0\",\n",
      "  \"type_vocab_size\": 2,\n",
      "  \"use_cache\": true,\n",
      "  \"vocab_size\": 30522\n",
      "}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from transformers import *\n",
    "\n",
    "tokenizer = AutoTokenizer.from_pretrained('microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### preprocessing - truncation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "de8443ea-dcd0-4615-bd52-179f7fb7e200",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "def preprocess_function(examples):\n",
    "    return tokenizer(examples[\"text\"], truncation=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "2a64b0bd-9340-4915-9799-022eb8e987ce",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "c458d1f25bc04592a87f070613b56b32",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Map:   0%|          | 0/3350 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "67b23342e85b49a28f9d7e4acd4b718a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Map:   0%|          | 0/838 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tokenizer_dataset = dataset.map(preprocess_function, batched=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### preprocessing - padding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "f656a668-e8ed-49f2-8152-0add1f9ba585",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "from transformers import DataCollatorWithPadding\n",
    "\n",
    "data_collator = DataCollatorWithPadding(tokenizer=tokenizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Model training"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "a6948c48-6b7d-4697-b24c-0e98a67b678f",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4ca50cbcdf214f5a928091f263ced8a0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading builder script:   0%|          | 0.00/4.20k [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import evaluate\n",
    "\n",
    "accuracy = evaluate.load(\"accuracy\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "e0538722-79f1-43be-98ed-70c564a8283b",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "\n",
    "\n",
    "def compute_metrics(eval_pred):\n",
    "    predictions, labels = eval_pred\n",
    "    predictions = np.argmax(predictions, axis=1)\n",
    "    return accuracy.compute(predictions=predictions, references=labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "44661993-af0e-45e7-9682-bf4b16f13dbc",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "loading configuration file config.json from cache at /root/.cache/huggingface/hub/models--microsoft--BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext/snapshots/eaa409b6b7c9380a5f2ba7a59aa97712ff30f386/config.json\n",
      "Model config BertConfig {\n",
      "  \"_name_or_path\": \"microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext\",\n",
      "  \"architectures\": [\n",
      "    \"BertForMaskedLM\"\n",
      "  ],\n",
      "  \"attention_probs_dropout_prob\": 0.1,\n",
      "  \"classifier_dropout\": null,\n",
      "  \"hidden_act\": \"gelu\",\n",
      "  \"hidden_dropout_prob\": 0.1,\n",
      "  \"hidden_size\": 768,\n",
      "  \"id2label\": {\n",
      "    \"0\": \"created\",\n",
      "    \"1\": \"used\",\n",
      "    \"2\": \"mention\",\n",
      "    \"3\": \"none\"\n",
      "  },\n",
      "  \"initializer_range\": 0.02,\n",
      "  \"intermediate_size\": 3072,\n",
      "  \"label2id\": {\n",
      "    \"created\": 0,\n",
      "    \"mention\": 2,\n",
      "    \"none\": 3,\n",
      "    \"used\": 1\n",
      "  },\n",
      "  \"layer_norm_eps\": 1e-12,\n",
      "  \"max_position_embeddings\": 512,\n",
      "  \"model_type\": \"bert\",\n",
      "  \"num_attention_heads\": 12,\n",
      "  \"num_hidden_layers\": 12,\n",
      "  \"pad_token_id\": 0,\n",
      "  \"position_embedding_type\": \"absolute\",\n",
      "  \"transformers_version\": \"4.31.0\",\n",
      "  \"type_vocab_size\": 2,\n",
      "  \"use_cache\": true,\n",
      "  \"vocab_size\": 30522\n",
      "}\n",
      "\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a1253b8974694b708635c076f03ace97",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Downloading pytorch_model.bin:   0%|          | 0.00/440M [00:00<?, ?B/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "loading weights file pytorch_model.bin from cache at /root/.cache/huggingface/hub/models--microsoft--BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext/snapshots/eaa409b6b7c9380a5f2ba7a59aa97712ff30f386/pytorch_model.bin\n",
      "Some weights of the model checkpoint at microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext were not used when initializing BertForSequenceClassification: ['cls.predictions.transform.dense.bias', 'cls.seq_relationship.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.decoder.bias', 'cls.predictions.decoder.weight', 'cls.predictions.transform.dense.weight', 'cls.seq_relationship.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.bias']\n",
      "- This IS expected if you are initializing BertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).\n",
      "- This IS NOT expected if you are initializing BertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).\n",
      "Some weights of BertForSequenceClassification were not initialized from the model checkpoint at microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext and are newly initialized: ['classifier.bias', 'classifier.weight']\n",
      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n"
     ]
    }
   ],
   "source": [
    "from transformers import AutoModelForSequenceClassification, TrainingArguments, Trainer\n",
    "# model = BertForSequenceClassification.from_pretrained('allenai/scibert_scivocab_uncased', num_labels=4,)\n",
    "model = AutoModelForSequenceClassification.from_pretrained(\n",
    "    'microsoft/BiomedNLP-PubMedBERT-base-uncased-abstract-fulltext', num_labels=4, id2label=id2label, label2id=label2id\n",
    ")\n",
    "\n",
    "# model = AutoModelForSequenceClassification.from_pretrained(\n",
    "#     \"distilbert-base-uncased\", num_labels=4, id2label=id2label, label2id=label2id\n",
    "# )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "8387fbd0-bcfd-4fcb-a54a-b2da93ae8126",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BertForSequenceClassification(\n",
       "  (bert): BertModel(\n",
       "    (embeddings): BertEmbeddings(\n",
       "      (word_embeddings): Embedding(30522, 768, padding_idx=0)\n",
       "      (position_embeddings): Embedding(512, 768)\n",
       "      (token_type_embeddings): Embedding(2, 768)\n",
       "      (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "      (dropout): Dropout(p=0.1, inplace=False)\n",
       "    )\n",
       "    (encoder): BertEncoder(\n",
       "      (layer): ModuleList(\n",
       "        (0-11): 12 x BertLayer(\n",
       "          (attention): BertAttention(\n",
       "            (self): BertSelfAttention(\n",
       "              (query): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (key): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (value): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (dropout): Dropout(p=0.1, inplace=False)\n",
       "            )\n",
       "            (output): BertSelfOutput(\n",
       "              (dense): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "              (dropout): Dropout(p=0.1, inplace=False)\n",
       "            )\n",
       "          )\n",
       "          (intermediate): BertIntermediate(\n",
       "            (dense): Linear(in_features=768, out_features=3072, bias=True)\n",
       "            (intermediate_act_fn): GELUActivation()\n",
       "          )\n",
       "          (output): BertOutput(\n",
       "            (dense): Linear(in_features=3072, out_features=768, bias=True)\n",
       "            (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "            (dropout): Dropout(p=0.1, inplace=False)\n",
       "          )\n",
       "        )\n",
       "      )\n",
       "    )\n",
       "    (pooler): BertPooler(\n",
       "      (dense): Linear(in_features=768, out_features=768, bias=True)\n",
       "      (activation): Tanh()\n",
       "    )\n",
       "  )\n",
       "  (dropout): Dropout(p=0.1, inplace=False)\n",
       "  (classifier): Linear(in_features=768, out_features=4, bias=True)\n",
       ")"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "a838bfad-27a9-4365-b21a-39eba0f8d0dc",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "device(type='cpu')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.device"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "a20be7e7-67d4-4f5a-8eb8-6663c1db500d",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BertForSequenceClassification(\n",
       "  (bert): BertModel(\n",
       "    (embeddings): BertEmbeddings(\n",
       "      (word_embeddings): Embedding(30522, 768, padding_idx=0)\n",
       "      (position_embeddings): Embedding(512, 768)\n",
       "      (token_type_embeddings): Embedding(2, 768)\n",
       "      (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "      (dropout): Dropout(p=0.1, inplace=False)\n",
       "    )\n",
       "    (encoder): BertEncoder(\n",
       "      (layer): ModuleList(\n",
       "        (0-11): 12 x BertLayer(\n",
       "          (attention): BertAttention(\n",
       "            (self): BertSelfAttention(\n",
       "              (query): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (key): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (value): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (dropout): Dropout(p=0.1, inplace=False)\n",
       "            )\n",
       "            (output): BertSelfOutput(\n",
       "              (dense): Linear(in_features=768, out_features=768, bias=True)\n",
       "              (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "              (dropout): Dropout(p=0.1, inplace=False)\n",
       "            )\n",
       "          )\n",
       "          (intermediate): BertIntermediate(\n",
       "            (dense): Linear(in_features=768, out_features=3072, bias=True)\n",
       "            (intermediate_act_fn): GELUActivation()\n",
       "          )\n",
       "          (output): BertOutput(\n",
       "            (dense): Linear(in_features=3072, out_features=768, bias=True)\n",
       "            (LayerNorm): LayerNorm((768,), eps=1e-12, elementwise_affine=True)\n",
       "            (dropout): Dropout(p=0.1, inplace=False)\n",
       "          )\n",
       "        )\n",
       "      )\n",
       "    )\n",
       "    (pooler): BertPooler(\n",
       "      (dense): Linear(in_features=768, out_features=768, bias=True)\n",
       "      (activation): Tanh()\n",
       "    )\n",
       "  )\n",
       "  (dropout): Dropout(p=0.1, inplace=False)\n",
       "  (classifier): Linear(in_features=768, out_features=4, bias=True)\n",
       ")"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "device = 'cuda:0'\n",
    "model.to(device)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "d958e991-d0ea-40e5-a47d-1202d6615fc2",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "device(type='cuda', index=0)"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.device"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "364b786d-3657-4159-8fbb-153d46045e81",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found safetensors installation, but --save_safetensors=False. Safetensors should be a preferred weights saving format due to security and performance reasons. If your model cannot be saved by safetensors please feel free to open an issue at https://github.com/huggingface/safetensors!\n",
      "PyTorch: setting up devices\n",
      "The default value for the training argument `--report_to` will change in v5 (from all installed integrations to none). In v5, you will need to use `--report_to all` to get the same behavior as now. You should start updating your code and make this info disappear :-).\n",
      "The following columns in the training set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "/databricks/python/lib/python3.10/site-packages/transformers/optimization.py:411: FutureWarning: This implementation of AdamW is deprecated and will be removed in a future version. Use the PyTorch implementation torch.optim.AdamW instead, or set `no_deprecation_warning=True` to disable this warning\n",
      "  warnings.warn(\n",
      "***** Running training *****\n",
      "  Num examples = 3,350\n",
      "  Num Epochs = 20\n",
      "  Instantaneous batch size per device = 10\n",
      "  Total train batch size (w. parallel, distributed & accumulation) = 10\n",
      "  Gradient Accumulation steps = 1\n",
      "  Total optimization steps = 6,700\n",
      "  Number of trainable parameters = 109,485,316\n",
      "You're using a BertTokenizerFast tokenizer. Please note that with a fast tokenizer, using the `__call__` method is faster than using a method to encode the text followed by a call to the `pad` method to get a padded encoding.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "    <div>\n",
       "      \n",
       "      <progress value='6700' max='6700' style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
       "      [6700/6700 12:30, Epoch 20/20]\n",
       "    </div>\n",
       "    <table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       " <tr style=\"text-align: left;\">\n",
       "      <th>Epoch</th>\n",
       "      <th>Training Loss</th>\n",
       "      <th>Validation Loss</th>\n",
       "      <th>Accuracy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>No log</td>\n",
       "      <td>0.363615</td>\n",
       "      <td>0.884248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>0.378700</td>\n",
       "      <td>0.351738</td>\n",
       "      <td>0.903341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>0.182300</td>\n",
       "      <td>0.446255</td>\n",
       "      <td>0.912888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>0.182300</td>\n",
       "      <td>0.512782</td>\n",
       "      <td>0.906921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>0.087400</td>\n",
       "      <td>0.519526</td>\n",
       "      <td>0.911695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>0.046300</td>\n",
       "      <td>0.600673</td>\n",
       "      <td>0.909308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>0.046300</td>\n",
       "      <td>0.633182</td>\n",
       "      <td>0.910501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>0.024900</td>\n",
       "      <td>0.622202</td>\n",
       "      <td>0.916468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>0.008500</td>\n",
       "      <td>0.632987</td>\n",
       "      <td>0.910501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>0.008500</td>\n",
       "      <td>0.718445</td>\n",
       "      <td>0.905728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>0.004300</td>\n",
       "      <td>0.608050</td>\n",
       "      <td>0.922434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>0.001600</td>\n",
       "      <td>0.672015</td>\n",
       "      <td>0.921241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>0.001600</td>\n",
       "      <td>0.661155</td>\n",
       "      <td>0.921241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>0.001900</td>\n",
       "      <td>0.649473</td>\n",
       "      <td>0.921241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>0.003900</td>\n",
       "      <td>0.677775</td>\n",
       "      <td>0.918854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>0.003900</td>\n",
       "      <td>0.688295</td>\n",
       "      <td>0.920048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>0.695542</td>\n",
       "      <td>0.920048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.706867</td>\n",
       "      <td>0.917661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>0.000400</td>\n",
       "      <td>0.699225</td>\n",
       "      <td>0.918854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.694458</td>\n",
       "      <td>0.921241</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table><p>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "The following columns in the evaluation set don't have a corresponding argument in `BertForSequenceClassification.forward` and have been ignored: text. If text are not expected by `BertForSequenceClassification.forward`,  you can safely ignore this message.\n",
      "***** Running Evaluation *****\n",
      "  Num examples = 838\n",
      "  Batch size = 10\n",
      "\n",
      "\n",
      "Training completed. Do not forget to share your model on huggingface.co/models =)\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "TrainOutput(global_step=6700, training_loss=0.055254011510718445, metrics={'train_runtime': 751.9343, 'train_samples_per_second': 89.104, 'train_steps_per_second': 8.91, 'total_flos': 2464994012692480.0, 'train_loss': 0.055254011510718445, 'epoch': 20.0})"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "training_args = TrainingArguments(\n",
    "    output_dir=\"./tmp/\",\n",
    "    learning_rate=2e-5,\n",
    "    per_device_train_batch_size=10,\n",
    "    per_device_eval_batch_size=10,\n",
    "    num_train_epochs=20,\n",
    "    weight_decay=0.01,\n",
    "    evaluation_strategy=\"epoch\",\n",
    "    # load_best_model_at_end=True,\n",
    "    save_strategy=\"no\",\n",
    "    # push_to_hub=True,\n",
    "    \n",
    ")\n",
    "\n",
    "trainer = Trainer(\n",
    "    model=model,\n",
    "    args=training_args,\n",
    "    train_dataset=tokenizer_dataset[\"train\"],\n",
    "    eval_dataset=tokenizer_dataset[\"test\"],\n",
    "    tokenizer=tokenizer,\n",
    "    data_collator=data_collator,\n",
    "    compute_metrics=compute_metrics,\n",
    ")\n",
    "\n",
    "trainer.train()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validation on CZI dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "1e2373e7-364e-428a-a878-8ec06cfaf50f",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doi_link</th>\n",
       "      <th>source</th>\n",
       "      <th>text</th>\n",
       "      <th>software</th>\n",
       "      <th>intent</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://doi.org/10.1371/journal.pone.0043845</td>\n",
       "      <td>10. Confocal microscopy</td>\n",
       "      <td>Confocal laser scanning was performed on a Oly...</td>\n",
       "      <td>Fluoview</td>\n",
       "      <td>used</td>\n",
       "      <td>usage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>https://doi.org/10.1155/2014/395212</td>\n",
       "      <td>5. Simulation Results and Analysis</td>\n",
       "      <td>In Figure 11, p miss of SVM-SMP is nearly equa...</td>\n",
       "      <td>SVM</td>\n",
       "      <td>used</td>\n",
       "      <td>usage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://doi.org/10.3389/fphys.2020.587057</td>\n",
       "      <td>Network Architecture and Training Parameters</td>\n",
       "      <td>A pattern recognition network was created usin...</td>\n",
       "      <td>MATLAB</td>\n",
       "      <td>used</td>\n",
       "      <td>usage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://doi.org/10.1186/s13046-018-0997-7</td>\n",
       "      <td>paper_abstract</td>\n",
       "      <td>Finally, we investigated the potential molecul...</td>\n",
       "      <td>analysis (GSEA)</td>\n",
       "      <td>used</td>\n",
       "      <td>usage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://doi.org/10.7717/peerj.9470</td>\n",
       "      <td>Conclusions</td>\n",
       "      <td>The GitHub repository for this study has been ...</td>\n",
       "      <td>Python</td>\n",
       "      <td>used</td>\n",
       "      <td>usage</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>https://joss.theoj.org/papers/10.21105/joss.05556</td>\n",
       "      <td>title</td>\n",
       "      <td>PASCal Python: A Principal Axis Strain Calculator</td>\n",
       "      <td>PASCal Python</td>\n",
       "      <td>creation</td>\n",
       "      <td>creation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>https://joss.theoj.org/papers/10.21105/joss.05313</td>\n",
       "      <td>title</td>\n",
       "      <td>QuaC: A Pipeline Implementing Quality Control ...</td>\n",
       "      <td>QuaC</td>\n",
       "      <td>creation</td>\n",
       "      <td>creation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>https://joss.theoj.org/papers/10.21105/joss.05251</td>\n",
       "      <td>title</td>\n",
       "      <td>ReSurfEMG: A Python library for preprocessing ...</td>\n",
       "      <td>ReSurfEMG</td>\n",
       "      <td>creation</td>\n",
       "      <td>creation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>https://joss.theoj.org/papers/10.21105/joss.05562</td>\n",
       "      <td>title</td>\n",
       "      <td>DARTS: The Data Analysis Remote Treatment Service</td>\n",
       "      <td>DARTS</td>\n",
       "      <td>creation</td>\n",
       "      <td>creation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>https://joss.theoj.org/papers/10.21105/joss.05580</td>\n",
       "      <td>title</td>\n",
       "      <td>SMART: Spatial Modeling Algorithms for Reactio...</td>\n",
       "      <td>SMART</td>\n",
       "      <td>creation</td>\n",
       "      <td>creation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>410 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              doi_link  ...     label\n",
       "0         https://doi.org/10.1371/journal.pone.0043845  ...     usage\n",
       "1                  https://doi.org/10.1155/2014/395212  ...     usage\n",
       "2            https://doi.org/10.3389/fphys.2020.587057  ...     usage\n",
       "3            https://doi.org/10.1186/s13046-018-0997-7  ...     usage\n",
       "4                   https://doi.org/10.7717/peerj.9470  ...     usage\n",
       "..                                                 ...  ...       ...\n",
       "405  https://joss.theoj.org/papers/10.21105/joss.05556  ...  creation\n",
       "406  https://joss.theoj.org/papers/10.21105/joss.05313  ...  creation\n",
       "407  https://joss.theoj.org/papers/10.21105/joss.05251  ...  creation\n",
       "408  https://joss.theoj.org/papers/10.21105/joss.05562  ...  creation\n",
       "409  https://joss.theoj.org/papers/10.21105/joss.05580  ...  creation\n",
       "\n",
       "[410 rows x 6 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from datasets import Dataset\n",
    "import pandas as pd\n",
    "\n",
    "df_evaluate = pd.read_csv('../data/czi_val_merged.csv')\n",
    "df_evaluate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "42252d83-82d4-47bd-aec3-046a677aaadc",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "text = df_evaluate['text'].values.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "78b61f51-6f29-4c1c-a320-6ed8387a9615",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Disabling tokenizer parallelism, we're using DataLoader multithreading already\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[{'label': 'used', 'score': 0.9999843835830688},\n",
       " {'label': 'used', 'score': 0.6080527901649475},\n",
       " {'label': 'mention', 'score': 0.9999368190765381},\n",
       " {'label': 'used', 'score': 0.9999810457229614},\n",
       " {'label': 'created', 'score': 0.9999082088470459},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9336268901824951},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999831914901733},\n",
       " {'label': 'used', 'score': 0.999984622001648},\n",
       " {'label': 'used', 'score': 0.9999843835830688},\n",
       " {'label': 'used', 'score': 0.9999841451644897},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999925971031189},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9992075562477112},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9999369382858276},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'mention', 'score': 0.999933123588562},\n",
       " {'label': 'used', 'score': 0.9999833106994629},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999794960021973},\n",
       " {'label': 'used', 'score': 0.9916220903396606},\n",
       " {'label': 'used', 'score': 0.9999810457229614},\n",
       " {'label': 'used', 'score': 0.9889782071113586},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.999984622001648},\n",
       " {'label': 'none', 'score': 0.996573805809021},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999847412109375},\n",
       " {'label': 'mention', 'score': 0.9999361038208008},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999830722808838},\n",
       " {'label': 'used', 'score': 0.6643899083137512},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999767541885376},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9999078512191772},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999845027923584},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999815225601196},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999817609786987},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9999356269836426},\n",
       " {'label': 'used', 'score': 0.999981164932251},\n",
       " {'label': 'none', 'score': 0.9999809265136719},\n",
       " {'label': 'used', 'score': 0.9999860525131226},\n",
       " {'label': 'used', 'score': 0.9999860525131226},\n",
       " {'label': 'used', 'score': 0.9999823570251465},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999834299087524},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999840259552002},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.999980092048645},\n",
       " {'label': 'used', 'score': 0.9999743700027466},\n",
       " {'label': 'used', 'score': 0.9999841451644897},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999786615371704},\n",
       " {'label': 'mention', 'score': 0.9997147917747498},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999669790267944},\n",
       " {'label': 'used', 'score': 0.9999830722808838},\n",
       " {'label': 'used', 'score': 0.9999815225601196},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999860525131226},\n",
       " {'label': 'used', 'score': 0.9999843835830688},\n",
       " {'label': 'used', 'score': 0.9999837875366211},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'mention', 'score': 0.9985747337341309},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999837875366211},\n",
       " {'label': 'used', 'score': 0.999984860420227},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999780654907227},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999839067459106},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9998201727867126},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999819993972778},\n",
       " {'label': 'used', 'score': 0.9999821186065674},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'mention', 'score': 0.9999359846115112},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'mention', 'score': 0.9999330043792725},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999463558197021},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999833106994629},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999783039093018},\n",
       " {'label': 'used', 'score': 0.998197615146637},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.999983549118042},\n",
       " {'label': 'used', 'score': 0.9999839067459106},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999847412109375},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999821186065674},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999984860420227},\n",
       " {'label': 'used', 'score': 0.9920112490653992},\n",
       " {'label': 'used', 'score': 0.9849387407302856},\n",
       " {'label': 'used', 'score': 0.9999823570251465},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9999377727508545},\n",
       " {'label': 'mention', 'score': 0.9999372959136963},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9987817406654358},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999841451644897},\n",
       " {'label': 'mention', 'score': 0.9999374151229858},\n",
       " {'label': 'used', 'score': 0.9999817609786987},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.999196469783783},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'mention', 'score': 0.99017333984375},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'none', 'score': 0.9999021291732788},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999815225601196},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999840259552002},\n",
       " {'label': 'used', 'score': 0.9999841451644897},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.999984860420227},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'mention', 'score': 0.9998080134391785},\n",
       " {'label': 'none', 'score': 0.9999810457229614},\n",
       " {'label': 'mention', 'score': 0.9999330043792725},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.999972939491272},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'none', 'score': 0.9990150928497314},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.999984622001648},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'mention', 'score': 0.9976494908332825},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999984860420227},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999840259552002},\n",
       " {'label': 'used', 'score': 0.9999843835830688},\n",
       " {'label': 'used', 'score': 0.6080527901649475},\n",
       " {'label': 'mention', 'score': 0.9999368190765381},\n",
       " {'label': 'used', 'score': 0.9999810457229614},\n",
       " {'label': 'created', 'score': 0.9999082088470459},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9336268901824951},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999831914901733},\n",
       " {'label': 'used', 'score': 0.999984622001648},\n",
       " {'label': 'used', 'score': 0.9999843835830688},\n",
       " {'label': 'used', 'score': 0.9999841451644897},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999925971031189},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9992075562477112},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9999369382858276},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'mention', 'score': 0.999933123588562},\n",
       " {'label': 'used', 'score': 0.9999833106994629},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999794960021973},\n",
       " {'label': 'used', 'score': 0.9916220903396606},\n",
       " {'label': 'used', 'score': 0.9999810457229614},\n",
       " {'label': 'used', 'score': 0.9889782071113586},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.999984622001648},\n",
       " {'label': 'none', 'score': 0.996573805809021},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999847412109375},\n",
       " {'label': 'mention', 'score': 0.9999361038208008},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999830722808838},\n",
       " {'label': 'used', 'score': 0.6643899083137512},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999767541885376},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9999078512191772},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999845027923584},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999815225601196},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999817609786987},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9999356269836426},\n",
       " {'label': 'used', 'score': 0.999981164932251},\n",
       " {'label': 'none', 'score': 0.9999809265136719},\n",
       " {'label': 'used', 'score': 0.9999860525131226},\n",
       " {'label': 'used', 'score': 0.9999860525131226},\n",
       " {'label': 'used', 'score': 0.9999823570251465},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999834299087524},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999840259552002},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.999980092048645},\n",
       " {'label': 'used', 'score': 0.9999743700027466},\n",
       " {'label': 'used', 'score': 0.9999841451644897},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999786615371704},\n",
       " {'label': 'mention', 'score': 0.9997147917747498},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999669790267944},\n",
       " {'label': 'used', 'score': 0.9999830722808838},\n",
       " {'label': 'used', 'score': 0.9999815225601196},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999860525131226},\n",
       " {'label': 'used', 'score': 0.9999843835830688},\n",
       " {'label': 'used', 'score': 0.9999837875366211},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'mention', 'score': 0.9985747337341309},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999837875366211},\n",
       " {'label': 'used', 'score': 0.999984860420227},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999780654907227},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999839067459106},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9998201727867126},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999819993972778},\n",
       " {'label': 'used', 'score': 0.9999821186065674},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'mention', 'score': 0.9999359846115112},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'mention', 'score': 0.9999330043792725},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999463558197021},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999833106994629},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999783039093018},\n",
       " {'label': 'used', 'score': 0.998197615146637},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.999983549118042},\n",
       " {'label': 'used', 'score': 0.9999839067459106},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999847412109375},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999821186065674},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999853372573853},\n",
       " {'label': 'used', 'score': 0.9999850988388062},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999984860420227},\n",
       " {'label': 'used', 'score': 0.9920112490653992},\n",
       " {'label': 'used', 'score': 0.9849387407302856},\n",
       " {'label': 'used', 'score': 0.9999823570251465},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9999377727508545},\n",
       " {'label': 'mention', 'score': 0.9999372959136963},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'mention', 'score': 0.9987817406654358},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999841451644897},\n",
       " {'label': 'mention', 'score': 0.9999374151229858},\n",
       " {'label': 'used', 'score': 0.9999817609786987},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.999196469783783},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'mention', 'score': 0.99017333984375},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'none', 'score': 0.9999021291732788},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'used', 'score': 0.9999852180480957},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999815225601196},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999840259552002},\n",
       " {'label': 'used', 'score': 0.9999841451644897},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.999984860420227},\n",
       " {'label': 'used', 'score': 0.9999849796295166},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'mention', 'score': 0.9998080134391785},\n",
       " {'label': 'none', 'score': 0.9999810457229614},\n",
       " {'label': 'mention', 'score': 0.9999330043792725},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.9999854564666748},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.999972939491272},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'none', 'score': 0.9990150928497314},\n",
       " {'label': 'used', 'score': 0.9999856948852539},\n",
       " {'label': 'used', 'score': 0.999984622001648},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999985933303833},\n",
       " {'label': 'mention', 'score': 0.9976494908332825},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.999984860420227},\n",
       " {'label': 'used', 'score': 0.9999858140945435},\n",
       " {'label': 'used', 'score': 0.9999855756759644},\n",
       " {'label': 'used', 'score': 0.9999840259552002},\n",
       " {'label': 'none', 'score': 0.9998348951339722},\n",
       " {'label': 'mention', 'score': 0.9999388456344604},\n",
       " {'label': 'mention', 'score': 0.9999362230300903},\n",
       " {'label': 'mention', 'score': 0.999931812286377},\n",
       " {'label': 'mention', 'score': 0.999937891960144},\n",
       " {'label': 'mention', 'score': 0.9999345541000366},\n",
       " {'label': 'mention', 'score': 0.9999361038208008},\n",
       " {'label': 'mention', 'score': 0.9999369382858276},\n",
       " {'label': 'mention', 'score': 0.9999207258224487},\n",
       " {'label': 'mention', 'score': 0.9999310970306396}]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from transformers import pipeline\n",
    "\n",
    "model = model.to('cpu')\n",
    "classifier = pipeline(\"sentiment-analysis\", model=model, tokenizer=tokenizer)\n",
    "classifier(text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "e06a3606-e6f5-4097-8d57-99ae2c3b567d",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/root/.ipykernel/2327/command-2884248819784294-2481495275:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_evaluate['label'] = df_evaluate['label'].apply(lambda x: map_label[x])\n"
     ]
    }
   ],
   "source": [
    "map_label = {'usage':'used', 'creation':'created', 'none': 'none', 'mention': 'mention'}\n",
    "df_evaluate = df_evaluate[['text', 'label']]\n",
    "df_evaluate['label'] = df_evaluate['label'].apply(lambda x: map_label[x])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "b30711bd-b6a8-41e0-b30f-7ed55f1aae75",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>text</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Confocal laser scanning was performed on a Oly...</td>\n",
       "      <td>used</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>In Figure 11, p miss of SVM-SMP is nearly equa...</td>\n",
       "      <td>used</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A pattern recognition network was created usin...</td>\n",
       "      <td>used</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Finally, we investigated the potential molecul...</td>\n",
       "      <td>used</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>The GitHub repository for this study has been ...</td>\n",
       "      <td>used</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>PASCal Python: A Principal Axis Strain Calculator</td>\n",
       "      <td>created</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>QuaC: A Pipeline Implementing Quality Control ...</td>\n",
       "      <td>created</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>ReSurfEMG: A Python library for preprocessing ...</td>\n",
       "      <td>created</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>DARTS: The Data Analysis Remote Treatment Service</td>\n",
       "      <td>created</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>SMART: Spatial Modeling Algorithms for Reactio...</td>\n",
       "      <td>created</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>410 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  text    label\n",
       "0    Confocal laser scanning was performed on a Oly...     used\n",
       "1    In Figure 11, p miss of SVM-SMP is nearly equa...     used\n",
       "2    A pattern recognition network was created usin...     used\n",
       "3    Finally, we investigated the potential molecul...     used\n",
       "4    The GitHub repository for this study has been ...     used\n",
       "..                                                 ...      ...\n",
       "405  PASCal Python: A Principal Axis Strain Calculator  created\n",
       "406  QuaC: A Pipeline Implementing Quality Control ...  created\n",
       "407  ReSurfEMG: A Python library for preprocessing ...  created\n",
       "408  DARTS: The Data Analysis Remote Treatment Service  created\n",
       "409  SMART: Spatial Modeling Algorithms for Reactio...  created\n",
       "\n",
       "[410 rows x 2 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_evaluate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "e78a8736-7ee3-4969-9d78-12ac69796b74",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\r  0%|          | 0/410 [00:00<?, ?it/s]\r100%|██████████| 410/410 [00:00<00:00, 47070.25it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--------------------------------------------------------------------------------\n",
      "[text] A pattern recognition network was created using MATLAB r2017b to study the performance of the extracted gait parameters [label]: used\n",
      "[model label]: mention [pred score]: 0.9999368190765381\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The GitHub repository for this study has been created, including the Python scripts and the datasets used in this study (see https://github.com/smiile8888/rna-splice-sites-recognition) [label]: used\n",
      "[model label]: created [pred score]: 0.9999082088470459\n",
      "--------------------------------------------------------------------------------\n",
      "[text] ROI region of interest, STD standard deviation, w/o ROI detection landmark prediction without ROI detection, Mask R-CNN end-to-end model for ROI detection and landmark prediction with Mask R-CNN, w/ ROI detection (segmentation/fixed size or segmentation/variable size or regression /variable size) Landmark prediction with ROI detection of fixed or variable size. [label]: none\n",
      "[model label]: used [pred score]: 0.9992075562477112\n",
      "--------------------------------------------------------------------------------\n",
      "[text] After the creation of the true color and near infrared orthophotos, they were loaded into Matlab software (version 12) for further processing [label]: used\n",
      "[model label]: mention [pred score]: 0.9999369382858276\n",
      "--------------------------------------------------------------------------------\n",
      "[text] We then developed a lidar-derived AGB model using the nonlinear least squares “nls” function in R [24], based on a total of 43 1 ha plots (15 real 1 ha plots, and 28 aggregated ones, Additional file 1: Table S1) [label]: used\n",
      "[model label]: mention [pred score]: 0.999933123588562\n",
      "--------------------------------------------------------------------------------\n",
      "[text] pyProGA - A PyMOL plugin for protein residue network analysis [label]: none\n",
      "[model label]: used [pred score]: 0.9889782071113586\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Absorption correction: multi-scan (SADABS; Sheldrick, 1996 ▶) T min = 0.66, T max = 0.87 [label]: none\n",
      "[model label]: used [pred score]: 0.999984622001648\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Editors of journals that focus on meta‐analyses could also consider making statcheck a standard part of their peer review process (following the journals Psychological Science and the Journal of Experimental Social Psychology). [label]: mention\n",
      "[model label]: none [pred score]: 0.996573805809021\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Analyses were conducted using R version 4.0.3 (17) [label]: used\n",
      "[model label]: mention [pred score]: 0.9999361038208008\n",
      "--------------------------------------------------------------------------------\n",
      "[text] R scripts [label]: none\n",
      "[model label]: mention [pred score]: 0.9999356269836426\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Identifying major effect QTL underlying single or multiple traits in various populations determines the successful use of QTL through MAS [8] [label]: mention\n",
      "[model label]: none [pred score]: 0.9999809265136719\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Absorption correction: multi-scan (SADABS; Sheldrick, 1996 ▶) T min = 0.971, T max = 0.972 [label]: none\n",
      "[model label]: used [pred score]: 0.9999830722808838\n",
      "--------------------------------------------------------------------------------\n",
      "[text] By using a probabilistic method, PSORTb 3.0 outperforms CELLO, Cell-PLoc, SLP-Local, and the previous versions of the same tool [label]: used\n",
      "[model label]: mention [pred score]: 0.9985747337341309\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Statistical calculations were done using built-in functions in Matlab R2018a (The MathWorks, Inc). [label]: used\n",
      "[model label]: mention [pred score]: 0.9999359846115112\n",
      "--------------------------------------------------------------------------------\n",
      "[text] R: A language and environment for statistical computing [label]: none\n",
      "[model label]: mention [pred score]: 0.9999330043792725\n",
      "--------------------------------------------------------------------------------\n",
      "[text] This is most often done using heuristics found in utilities such as CLUSTAL W [18], MUSCLE [19], T-COFFEE [20], and ProbCons [21] [label]: mention\n",
      "[model label]: used [pred score]: 0.9999463558197021\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The six methods are: RDP [47], GENECONV [48], Bootscan [49], MaxChi [50], Chimaera [51], and SiScan [52] [label]: none\n",
      "[model label]: used [pred score]: 0.9999783039093018\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Data import from flat files is predominantly based on BioJava [24] which is used to parse multi-FASTA, EMBL [25], Genbank [12], and UniProt [11] formats [label]: mention\n",
      "[model label]: used [pred score]: 0.998197615146637\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The PhastCons program uses genome alignments across 30 vertebrate species (30-way) to identify conserved genomic regions based upon a phylogenetic hidden Markov model [97]. [label]: mention\n",
      "[model label]: used [pred score]: 0.9999839067459106\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The Affymetrix GCOS operating system controls the scanner and data acquisition functions [label]: mention\n",
      "[model label]: used [pred score]: 0.9920112490653992\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The major difference between Pocket-Finder and Q-SiteFinder is that in clustering in Pocket-Finder the sites produced by the Pocket-Finder are ranked according to the number of probes in the site rather than by probe energy [label]: mention\n",
      "[model label]: used [pred score]: 0.9849387407302856\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Community composition summaries (phylum and class levels) and the list of the top 30 OTUs (representing at least 20% of all reads in at least one of the samples) were plotted in R as a barplot and heatmap, respectively [label]: used\n",
      "[model label]: mention [pred score]: 0.9999377727508545\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The GeoJSON files, which can be used to test the performance of JSTS, are available in Datasets 1–10. [label]: used\n",
      "[model label]: mention [pred score]: 0.9999372959136963\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The LabVIEW-based software was used to define the scanning parameters (area to scan, frequency sweep configuration…) and load them into the Arduino board [label]: used\n",
      "[model label]: mention [pred score]: 0.9987817406654358\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Because in silico prediction tools may give different result, a good rule is to consider a prediction only when Polyphen2 (http//genetics.bwh.harvard.edu/pph2/) SIFT (http//sift.jcvi.org/), and the CONsensus DELeteriousness score of missense mutations (Condel) disease prediction tools give the same result [label]: mention\n",
      "[model label]: used [pred score]: 0.999196469783783\n",
      "--------------------------------------------------------------------------------\n",
      "[text] We found that applying the eFAST approach for a complex multi-compartment, multi-dataset, multi-metabolite PBPK model required less than 10 min of computational time on current hardware [label]: used\n",
      "[model label]: mention [pred score]: 0.99017333984375\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Arterial stiffness was measured using central (carotid– femoral) pulse wave velocity (aPWV) using a semi-automated device and software (SphygmoCor, AtCor Medical, Sydney, Australia) as previously described [39] [label]: used\n",
      "[model label]: none [pred score]: 0.9999021291732788\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Within LiST however the risks for mortality vis-à-vis stunting are defined in terms of the following four bands on the international normal distribution for height for age: [label]: mention\n",
      "[model label]: none [pred score]: 0.9999810457229614\n",
      "--------------------------------------------------------------------------------\n",
      "[text] R: A language and environment for statistical computing [label]: none\n",
      "[model label]: mention [pred score]: 0.9999330043792725\n",
      "--------------------------------------------------------------------------------\n",
      "[text] DCM, therefore, can also be regarded as more general, with the general linear model analysis being a specific situation in which the interaction parameters (first and second sets) are assumed to be zero. [label]: mention\n",
      "[model label]: none [pred score]: 0.9990150928497314\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Parameter configurations for XCMS, MZmine, and MAVEN. [label]: none\n",
      "[model label]: mention [pred score]: 0.9976494908332825\n",
      "--------------------------------------------------------------------------------\n",
      "[text] A pattern recognition network was created using MATLAB r2017b to study the performance of the extracted gait parameters [label]: used\n",
      "[model label]: mention [pred score]: 0.9999368190765381\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The GitHub repository for this study has been created, including the Python scripts and the datasets used in this study (see https://github.com/smiile8888/rna-splice-sites-recognition) [label]: used\n",
      "[model label]: created [pred score]: 0.9999082088470459\n",
      "--------------------------------------------------------------------------------\n",
      "[text] ROI region of interest, STD standard deviation, w/o ROI detection landmark prediction without ROI detection, Mask R-CNN end-to-end model for ROI detection and landmark prediction with Mask R-CNN, w/ ROI detection (segmentation/fixed size or segmentation/variable size or regression /variable size) Landmark prediction with ROI detection of fixed or variable size. [label]: none\n",
      "[model label]: used [pred score]: 0.9992075562477112\n",
      "--------------------------------------------------------------------------------\n",
      "[text] After the creation of the true color and near infrared orthophotos, they were loaded into Matlab software (version 12) for further processing [label]: used\n",
      "[model label]: mention [pred score]: 0.9999369382858276\n",
      "--------------------------------------------------------------------------------\n",
      "[text] We then developed a lidar-derived AGB model using the nonlinear least squares “nls” function in R [24], based on a total of 43 1 ha plots (15 real 1 ha plots, and 28 aggregated ones, Additional file 1: Table S1) [label]: used\n",
      "[model label]: mention [pred score]: 0.999933123588562\n",
      "--------------------------------------------------------------------------------\n",
      "[text] pyProGA - A PyMOL plugin for protein residue network analysis [label]: none\n",
      "[model label]: used [pred score]: 0.9889782071113586\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Absorption correction: multi-scan (SADABS; Sheldrick, 1996 ▶) T min = 0.66, T max = 0.87 [label]: none\n",
      "[model label]: used [pred score]: 0.999984622001648\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Editors of journals that focus on meta‐analyses could also consider making statcheck a standard part of their peer review process (following the journals Psychological Science and the Journal of Experimental Social Psychology). [label]: mention\n",
      "[model label]: none [pred score]: 0.996573805809021\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Analyses were conducted using R version 4.0.3 (17) [label]: used\n",
      "[model label]: mention [pred score]: 0.9999361038208008\n",
      "--------------------------------------------------------------------------------\n",
      "[text] R scripts [label]: none\n",
      "[model label]: mention [pred score]: 0.9999356269836426\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Identifying major effect QTL underlying single or multiple traits in various populations determines the successful use of QTL through MAS [8] [label]: mention\n",
      "[model label]: none [pred score]: 0.9999809265136719\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Absorption correction: multi-scan (SADABS; Sheldrick, 1996 ▶) T min = 0.971, T max = 0.972 [label]: none\n",
      "[model label]: used [pred score]: 0.9999830722808838\n",
      "--------------------------------------------------------------------------------\n",
      "[text] By using a probabilistic method, PSORTb 3.0 outperforms CELLO, Cell-PLoc, SLP-Local, and the previous versions of the same tool [label]: used\n",
      "[model label]: mention [pred score]: 0.9985747337341309\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Statistical calculations were done using built-in functions in Matlab R2018a (The MathWorks, Inc). [label]: used\n",
      "[model label]: mention [pred score]: 0.9999359846115112\n",
      "--------------------------------------------------------------------------------\n",
      "[text] R: A language and environment for statistical computing [label]: none\n",
      "[model label]: mention [pred score]: 0.9999330043792725\n",
      "--------------------------------------------------------------------------------\n",
      "[text] This is most often done using heuristics found in utilities such as CLUSTAL W [18], MUSCLE [19], T-COFFEE [20], and ProbCons [21] [label]: mention\n",
      "[model label]: used [pred score]: 0.9999463558197021\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The six methods are: RDP [47], GENECONV [48], Bootscan [49], MaxChi [50], Chimaera [51], and SiScan [52] [label]: none\n",
      "[model label]: used [pred score]: 0.9999783039093018\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Data import from flat files is predominantly based on BioJava [24] which is used to parse multi-FASTA, EMBL [25], Genbank [12], and UniProt [11] formats [label]: mention\n",
      "[model label]: used [pred score]: 0.998197615146637\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The PhastCons program uses genome alignments across 30 vertebrate species (30-way) to identify conserved genomic regions based upon a phylogenetic hidden Markov model [97]. [label]: mention\n",
      "[model label]: used [pred score]: 0.9999839067459106\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The Affymetrix GCOS operating system controls the scanner and data acquisition functions [label]: mention\n",
      "[model label]: used [pred score]: 0.9920112490653992\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The major difference between Pocket-Finder and Q-SiteFinder is that in clustering in Pocket-Finder the sites produced by the Pocket-Finder are ranked according to the number of probes in the site rather than by probe energy [label]: mention\n",
      "[model label]: used [pred score]: 0.9849387407302856\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Community composition summaries (phylum and class levels) and the list of the top 30 OTUs (representing at least 20% of all reads in at least one of the samples) were plotted in R as a barplot and heatmap, respectively [label]: used\n",
      "[model label]: mention [pred score]: 0.9999377727508545\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The GeoJSON files, which can be used to test the performance of JSTS, are available in Datasets 1–10. [label]: used\n",
      "[model label]: mention [pred score]: 0.9999372959136963\n",
      "--------------------------------------------------------------------------------\n",
      "[text] The LabVIEW-based software was used to define the scanning parameters (area to scan, frequency sweep configuration…) and load them into the Arduino board [label]: used\n",
      "[model label]: mention [pred score]: 0.9987817406654358\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Because in silico prediction tools may give different result, a good rule is to consider a prediction only when Polyphen2 (http//genetics.bwh.harvard.edu/pph2/) SIFT (http//sift.jcvi.org/), and the CONsensus DELeteriousness score of missense mutations (Condel) disease prediction tools give the same result [label]: mention\n",
      "[model label]: used [pred score]: 0.999196469783783\n",
      "--------------------------------------------------------------------------------\n",
      "[text] We found that applying the eFAST approach for a complex multi-compartment, multi-dataset, multi-metabolite PBPK model required less than 10 min of computational time on current hardware [label]: used\n",
      "[model label]: mention [pred score]: 0.99017333984375\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Arterial stiffness was measured using central (carotid– femoral) pulse wave velocity (aPWV) using a semi-automated device and software (SphygmoCor, AtCor Medical, Sydney, Australia) as previously described [39] [label]: used\n",
      "[model label]: none [pred score]: 0.9999021291732788\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Within LiST however the risks for mortality vis-à-vis stunting are defined in terms of the following four bands on the international normal distribution for height for age: [label]: mention\n",
      "[model label]: none [pred score]: 0.9999810457229614\n",
      "--------------------------------------------------------------------------------\n",
      "[text] R: A language and environment for statistical computing [label]: none\n",
      "[model label]: mention [pred score]: 0.9999330043792725\n",
      "--------------------------------------------------------------------------------\n",
      "[text] DCM, therefore, can also be regarded as more general, with the general linear model analysis being a specific situation in which the interaction parameters (first and second sets) are assumed to be zero. [label]: mention\n",
      "[model label]: none [pred score]: 0.9990150928497314\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Parameter configurations for XCMS, MZmine, and MAVEN. [label]: none\n",
      "[model label]: mention [pred score]: 0.9976494908332825\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Curifactory: A research experiment manager [label]: created\n",
      "[model label]: none [pred score]: 0.9998348951339722\n",
      "--------------------------------------------------------------------------------\n",
      "[text] libscientific: A Powerful C Library for Multivariate Analysis [label]: created\n",
      "[model label]: mention [pred score]: 0.9999388456344604\n",
      "--------------------------------------------------------------------------------\n",
      "[text] baseflow: a MATLAB and GNU Octave package for baseflow recession analysis [label]: created\n",
      "[model label]: mention [pred score]: 0.9999362230300903\n",
      "--------------------------------------------------------------------------------\n",
      "[text] gesel: a JavaScript package for client-side gene set enrichment [label]: created\n",
      "[model label]: mention [pred score]: 0.999931812286377\n",
      "--------------------------------------------------------------------------------\n",
      "[text] Qiskit Dynamics: A Python package for simulating the time dynamics of quantum systems [label]: created\n",
      "[model label]: mention [pred score]: 0.999937891960144\n",
      "--------------------------------------------------------------------------------\n",
      "[text] PASCal Python: A Principal Axis Strain Calculator [label]: created\n",
      "[model label]: mention [pred score]: 0.9999345541000366\n",
      "--------------------------------------------------------------------------------\n",
      "[text] QuaC: A Pipeline Implementing Quality Control Best Practices for Genome Sequencing and Exome Sequencing Data [label]: created\n",
      "[model label]: mention [pred score]: 0.9999361038208008\n",
      "--------------------------------------------------------------------------------\n",
      "[text] ReSurfEMG: A Python library for preprocessing and analysis of respiratory EMG. [label]: created\n",
      "[model label]: mention [pred score]: 0.9999369382858276\n",
      "--------------------------------------------------------------------------------\n",
      "[text] DARTS: The Data Analysis Remote Treatment Service [label]: created\n",
      "[model label]: mention [pred score]: 0.9999207258224487\n",
      "--------------------------------------------------------------------------------\n",
      "[text] SMART: Spatial Modeling Algorithms for Reactions and Transport [label]: created\n",
      "[model label]: mention [pred score]: 0.9999310970306396\n",
      "--------------------------------------------------------------------------------\n",
      "correct: 338 total: 410\n",
      "accuracy: 0.824390243902439\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "import tqdm\n",
    "cnt = 0\n",
    "classifier_list = classifier(text)\n",
    "for i in tqdm.tqdm(range(len(classifier_list))):\n",
    "    if classifier_list[i]['label'] == df_evaluate.loc[i, 'label']:\n",
    "        cnt += 1\n",
    "    else:\n",
    "        print('-'*80)\n",
    "        print('[text]', df_evaluate.loc[i, 'text'], '[label]:', df_evaluate.loc[i, 'label'])\n",
    "        print('[model label]:', classifier_list[i]['label'], '[pred score]:', classifier_list[i]['score'])\n",
    "print('-'*80)\n",
    "print('correct:', cnt, 'total:', len(df_evaluate))\n",
    "print('accuracy:', cnt/len(df_evaluate))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "49193aae-3db4-4fc6-ae83-2492238e90b5",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "from sklearn.metrics import precision_score, recall_score, f1_score, multilabel_confusion_matrix, accuracy_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "7a40afa6-2573-498e-9b08-a8e68377cee4",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "precision: 0.27881944444444445\n",
      "recall: 0.30447941888619856\n",
      "f1: 0.2885055687065854\n",
      "accuracy: 0.824390243902439\n"
     ]
    }
   ],
   "source": [
    "classifier_list = classifier(text)\n",
    "true_labels = [label2id[i] for i in df_evaluate['label'].values.tolist()]\n",
    "predicted_labels = [label2id[i['label']] for i in classifier_list]\n",
    "\n",
    "precision = precision_score(true_labels, predicted_labels, average='macro')\n",
    "recall = recall_score(true_labels, predicted_labels, average='macro')\n",
    "f1 = f1_score(true_labels, predicted_labels, average='macro')\n",
    "accuracy = accuracy_score(true_labels, predicted_labels)\n",
    "print('precision:', precision)\n",
    "print('recall:', recall)\n",
    "print('f1:', f1)\n",
    "print('accuracy:', accuracy)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Validation on CZI dataset - each label"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "57f37780-77e3-4b3c-b3f4-d149fdf5762b",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[[398,   2],\n",
       "        [ 10,   0]],\n",
       "\n",
       "       [[ 34,  22],\n",
       "        [ 24, 330]],\n",
       "\n",
       "       [[345,  37],\n",
       "        [ 20,   8]],\n",
       "\n",
       "       [[381,  11],\n",
       "        [ 18,   0]]])"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "confusion_matrices = multilabel_confusion_matrix(true_labels, predicted_labels)\n",
    "confusion_matrices\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "ac5e778a-fa82-402a-8464-1b978425fadf",
     "showTitle": false,
     "title": ""
    }
   },
   "source": [
    "Confusion matrix:\n",
    "[[TN, FP]\n",
    " [FN, TP]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "5950280a-014a-4281-bd65-f23e08c84727",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--------------------------------------------------------------------------------\n",
      "Label: created\n",
      "Precision: 0.0\n",
      "Recall: 0.0\n",
      "F1 Score: 0.0\n",
      "Confusion Matrix:\n",
      "[[398   2]\n",
      " [ 10   0]]\n",
      "--------------------------------------------------------------------------------\n",
      "Label: used\n",
      "Precision: 0.9375\n",
      "Recall: 0.9322033898305084\n",
      "F1 Score: 0.9348441926345609\n",
      "Confusion Matrix:\n",
      "[[ 34  22]\n",
      " [ 24 330]]\n",
      "--------------------------------------------------------------------------------\n",
      "Label: mention\n",
      "Precision: 0.17777777777777778\n",
      "Recall: 0.2857142857142857\n",
      "F1 Score: 0.2191780821917808\n",
      "Confusion Matrix:\n",
      "[[345  37]\n",
      " [ 20   8]]\n",
      "--------------------------------------------------------------------------------\n",
      "Label: none\n",
      "Precision: 0.0\n",
      "Recall: 0.0\n",
      "F1 Score: 0.0\n",
      "Confusion Matrix:\n",
      "[[381  11]\n",
      " [ 18   0]]\n"
     ]
    }
   ],
   "source": [
    "# Print the results\n",
    "for label in range(len(label2id)):\n",
    "    print('-'*80)\n",
    "    print(f\"Label: {id2label[label]}\")\n",
    "    print(\"Precision:\", precision[label])\n",
    "    print(\"Recall:\", recall[label])\n",
    "    print(\"F1 Score:\", f1[label])\n",
    "    print(\"Confusion Matrix:\")\n",
    "    print(confusion_matrices[label])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validation on test dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "46b8b9a1-7efd-4ab8-9edd-d387ad4cadb9",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['However, data from Haiti indicate that only 1/3 of those who receive PEP are fully compliant [19]. ',\n",
       " 'illuminaio provides the first open-source package for this task, and will promote wider uptake of the IDAT format as a standard for sharing Illumina BeadArray data in public databases, in the same way that the CEL file serves as the standard for the Affymetrix platform.\\n',\n",
       " 'The integration of CBFA within the OptFlux framework for ME enables the utilization of different model formats and standards and the integration with complementary methods for phenotype simulation and visualization of results.\\n',\n",
       " 'CoXpress is released as a package for R.\\n',\n",
       " 'Negative observed-minus-expected values signify that a color form was attacked less than expected and vice versa.',\n",
       " 'The male-to-female population ratios in age were almost constant from 2000 to 2009.',\n",
       " 'CoXpress is written in the native R language and has been fully tested on both windows and linux.\\n',\n",
       " 'The genomic locations of MER sequences were confirmed in the horse genome (Broad/equCab2), using UCSC Genome Browser (http://genome.ucsc.edu), and their sequences were analyzed to identify regions that overlapped with transcripts, using Ensembl 84 (http://www.ensembl.org).',\n",
       " 'We here applied the bootstrapping approach implemented in the SPM8 LI-toolbox, which is the current gold standard [36]. ',\n",
       " 'To address this challenge we have developed a new version of DREM termed interactive DREM (iDREM).\\n',\n",
       " 'The CeleST software can be used as a platform, and with application-specific tracking methods and measures, other cellular or animal behaviors can be analyzed (sperm mobility, zebrafish swimming, etc.) for any application that can use video (mating, sleeping, feeding, foraging, etc.).\\n',\n",
       " 'We used DAVID GO analysis software (http://david.abcc. ncifcrf.gov/, last accessed March 2013) to determine whether any functional categories were overrepresented at duplicated genes.',\n",
       " 'Statistical analysis, including the calculation of 95%-confidence intervals for bTB and MTBC prevalence, was performed using NCSS 2007 statistical software (Version 07.1.15; Kaysville, UT, USA).\\n\\n',\n",
       " 'Image preprocessing was carried out using SPM5 (http://www.fil.ion.ucl.ac.uk/spm) and Resting-State fMRI Data Analysis Toolkit (REST) (http:// www.restfmri.net).',\n",
       " 'Data and code are available at https://github.com/themusiclab/infant-vocal; the corpus is available at https://osf.io/m5yn2; the preregistration is at https://osf.io/5r72u; and readers may participate in the naïve listener experiment at https://themusiclab.org/quizzes/ids.',\n",
       " 'As cut off criteria in the BLAST searches we used a minimum alignment length of 100nt, E-value 0.001 a word size of 11, and a minimum of 70% sequence identity. ',\n",
       " 'Structural variants were identified using a de novo assembly-based approach using ABySS and Trans-ABySS (Simpson et al. 2009) and the alignment-based SV detection tool MOJO (Minimum Overlap Junction Optimizer; https://github.com/cband/MOJO).',\n",
       " 'Therefore, we normalized the taste responses by dividing the magnitudes of the responses to each taste stimuli by the response to 0.1 M NH4Cl, which is generally used as a standard stimulus in electrophysiological recordings of peripheral taste nerves.\\n\\n',\n",
       " 'There was slight variation between the TSE vacuolation profiles of the MM (CNS) and MM (spleen) in the VM mice which could indicate changes in titre.',\n",
       " 'Programming and online database \"OnlineTED\" was developed and programmed by students and lecturers at TUM.',\n",
       " 'LMEs were fitted using the lmer function in the lme4 package [72], and Levene’s tests were performed using the leveneTest function in the car package [73]. ',\n",
       " 'Molecular docking simulation of selected inhibitors against JAK2 was carried out using MOE [51].',\n",
       " 'Purified protein was run on size-exclusion Sephadex 200 chromatography columns to remove putative aggregates and eluted in 50 mM phosphate buffer (PB) (pH 8.0).',\n",
       " 'Microsoft Excel and Statistical Package for the Social Sciences (SPSS Version 22.0; Chicago, IL, USA) were used for all data analysis. ',\n",
       " 'fMRI can be defined as a technology that provides statistical maps of brain activation [6,7]. ',\n",
       " 'To achieve phase separation, samples were subjected to centrifugation at 12,000 g for 5 min at 4°C. ',\n",
       " 'The tuning process involved a nested cross-validation technique (5,000-fold, 5 times) using the caret package [45] on the training dataset to select the best parameters, such as the number of variables randomly sampled. ',\n",
       " 'Collection of socio-behavioral data using self-administered questionnaires (SAQ) was scheduled annually.',\n",
       " 'All analyses were conducted using Stata version 14.\\n\\n',\n",
       " 'This project was approved by the Research Ethics Committee of Beijing Normal University and the local education authorities (i.e., County Departments of Education) as well as the participating school principals. ',\n",
       " 'In both models, a land use intensity approach is used to relate the economic product with the respective area of industrial and commercial units. ',\n",
       " 'We collapsed the individual measures used for each personality trait into the first principal component (PC) score for each individual in the statistical package JMP, including scores from each trial for each individual. ',\n",
       " 'Physicochemical properties of the wild-type and new isoform were predicted by online tools ProtParam and ProtScale (http://www.expasy.ch/tools/ protscale.html), respectively.',\n",
       " 'All data fitting were carried out using OriginLab (Pro 8, OriginLab Corp., Northampton, MA).\\n\\n',\n",
       " 'All four DWI runs were merged and corrected for motion and eddy currents by registering each volume, including b0 volumes, to a reference DWI volume using FLIRT (Jenkinson and Smith 2001). ',\n",
       " 'All the analyses were conducted using SAS software (SAS Institute Inc., Cary, NC, USA). ',\n",
       " 'Study of these processes in detail promises to produce rewarding results both in basic and applied research.\\n\\n',\n",
       " 'Cortical nodes were defined in a standard cortical surface from FreeSurfer (fsaverage)(https://surfer.nmr.mgh.harvard.edu) and resampled into a regular grid using AFNI’s MapIcosahedron [42,43] with 10,242 nodes in each hemisphere. ',\n",
       " 'Paleoradiological evaluation as well as performance of multi-planar reconstructions (MPR) was carried out at the Picture Archiving and Communicating System (ImpaxEE, Agfa HealthCare, Bonn, Germany) in the Department of Radiology by the first author who is also experienced in paleoradiology and musculoskeletal radiology. ',\n",
       " 'All statistical analysis was performed using SigmaPlot (version 11).',\n",
       " 'All analyses were performed using Stata/SE 11.2 software for Mac(R).',\n",
       " 'GraphPad Prism 5.02 was used for statistical calculations.',\n",
       " 'Manual assembly of the obtained structures into a DEG7 trimer was performed using the PyMOL program (http://www.pymol.org/).',\n",
       " 'Statistical analyzes were performed using the STATA version 13 application (StataCorp, College Station, Texas, USA) and the thematic maps were constructed using the QGIS software version 2.12.3 (OSGeo, Beaverton, OR, USA). ',\n",
       " 'All data were entered into the IBM Statistical Package for the Social Sciences (SPSS) 18.',\n",
       " 'Data linkage was completed using Matlab.',\n",
       " 'The simulation of the pedigree and recalculation of LRS data was implemented with a purpose-specific programme coded in MATLAB (MathWorks), with animal model analyses implemented in AsReml (VSN International) as described above.\\n\\n',\n",
       " 'The sequences were analyzed for potential SSRs using Simple Sequence Repeat Identifi cation Tool (SSRIT) software ( Temnykh et al., 2001 ; http://www.gramene.org/db/markers/ssrtool/ ).',\n",
       " 'Criteria for excluding SNPs was:',\n",
       " 'To address these limitations, we present 4Cin, a method to generate 3D models and derive virtual Hi-C (vHi-C) heat maps of genomic loci based on 4C-seq or any kind of 4C-seq-like data, such as those derived from NG Capture-C.\\n',\n",
       " 'Comorbidity is calculated by TARN using the pre-existing medical conditions (PMC) score, modified from the Charlson Comorbidity Index. ',\n",
       " 'All of the PCR primers used in this study were designed using Primer-BLAST (http://blast.ncbi.nlm.nih.gov) and are shown in Table 1.',\n",
       " 'All analyses were conducted with Stata 14.1 using sampling weights and the suite of svy commands to accurately estimate standard errors.\\n\\n',\n",
       " '3.4.',\n",
       " 'Online responses were exported from an integrated research platform (Globalpark) by Opinion Matters personnel into a survey reporting software program (SNAP), and from thence were converted into an Excel spreadsheet. ',\n",
       " 'The applications we have implemented in ANDES allow an investigator to identify and compare the variations within a sample or among multiple samples.\\n',\n",
       " 'Statistical analyses were performed using SPSS v20 (Chicago, IL, USA). ',\n",
       " 'Wham and all associated software are covered under the MIT License and can be freely downloaded from github (https://github.com/zeeev/wham), with documentation on a wiki (http://zeeev.github.io/wham/).\\n',\n",
       " '18.0 (SPSS, Chicago, IL, USA) and SAS vers. ',\n",
       " 'Seeds were generated using the Wake Forest University PickAtlas [50, 51] using a 4mm radius.\\n\\n',\n",
       " 'These were recorded onto Datacam by the interventional cardiologist performing the procedure.\\n\\n',\n",
       " 'All GARLI analyses were performed with the default parameter settings and the GTR+gamma nucleotide substitution model.',\n",
       " 'The nucleotide sequences obtained were processed with the Phred/Phrap/ Consed package (http://www.phrap.org/phredphrapconsed .html).',\n",
       " 'The R-packages, tximport [52] and edgeR [53], were used to respectively summarize the expression at gene-level and normalize the data.',\n",
       " 'Table 1 provides summary information on the sociodemographic characteristics of counties.',\n",
       " 'Figure identifier and caption: 10.1371/journal.pone.0201364.g002 Geographic allocation of the 12 cities in the BTH urban agglomeration.',\n",
       " 'The consultant’s review workstation is a Macintosh computer running the medical imaging software OsiriX.',\n",
       " 'We present VisANT, an application for integrating biomolecular interaction data into a cohesive, graphical interface.\\n',\n",
       " 'Motivated by the challenge presented by the tremendous computational demand associated with epistatic QTL mapping, we conducted a thorough investigation of the model developed by Xu et al. and re-implemented the model algorithm using C/C++, resulting in the development of a web server-based tool named PEPIS (Pipeline for estimating EPIStatic genetic effects).\\n',\n",
       " 'MSACompro is an efficient and reliable multiple protein sequence alignment tool that can effectively incorporate predicted protein structural information into multiple sequence alignment.\\n',\n",
       " 'DLMs were implemented using the dlnm package version 2.2.6 in R Studio (Vienna, Austria) and other analyses were performed in SAS (v9.4, SAS Institute Inc., Cary, NC).\\n\\n',\n",
       " 'Chaste is written in C++, a compiled language that allows object-oriented class definitions.\\n',\n",
       " 'LCGM was performed in STATA (version 12.1, College Station, TX) using the traj procedure [24]. ',\n",
       " 'Compatible with any modern web browser except Internet Explorer (we tested it on Firefox v30.0, Chrome v36.0, Safari v6.1) Programming language: jQuery/HTML5, Python/Django, R Other requirements: A VirtualBox installation (for expansion and modification) License: GPL Any restrictions to use by non-academics: None\"\\n',\n",
       " 'These categories were retrieved from the DAVID database (https://david.ncifcrf.gov/) [35] including information from KEGG, PANTHER and GO_FAT annotations.',\n",
       " 'Analyses were carried out using Stata version 12.',\n",
       " 'Data analysis was performed using custom routines written in Python.',\n",
       " 'At present Chaste can only be used with Linux, although it works well via a Linux virtual machine (using software such as VirtualBox) on a host running Microsoft Windows or Mac OS X.\\n',\n",
       " 'These equations were implemented in a Matlab script that was validated by comparing its output to the tables provided in the ISO publication.',\n",
       " 'Our database, pseudoMap, aims at providing comprehensive resource for genome-wide identifying the functions and regulators of human TPGs.',\n",
       " 'All statistical analysis was conducted in Minitab 16 (Minitab Ltd).',\n",
       " 'Specifically, we used the frontal lobe mask as given by the WFU-Pickatlas as an inclusive mask to differentiate between regions of interest and regions of no interest. ',\n",
       " 'All statistical analyses were performed using SPSS (standard version 19.0, Chicago, IL). ',\n",
       " 'Codes developed from the qualitative data on motivators and quantitative data on demographic characteristics were entered in to SPSS version 21 for analysis.',\n",
       " 'Selected variants in Family A were tested for segregation using single-marker parametric linkage analysis based on the same parameters as the genome-wide multipoint linkage analysis, here using MERLIN [80] with customized bit size to accommodate the pedigree size. ',\n",
       " 'The data were analyzed with the immunoSEQ Analyzer (https://www.adaptivebiotech.com/immunoseq).',\n",
       " 'A dynamic, stochastic, agent-based model, written in NetLogo, was used to simulate the growth and mortality of a snail population, and infection by S. mansoni. ',\n",
       " 'In clinical trials, MRI is commonly used as a radiologic diagnostic aid to exclude individuals with incidental brain pathologies. ',\n",
       " 'This vector indicates the strength and direction of modulation for each individual neuron in the population, in response to a particular variable in a particular epoch. ',\n",
       " 'The results were analyzed with SigmaStat® 3.1 for Windows® (Systat Software Inc., Richmond, CA). ',\n",
       " 'All analyses were carried out in MATLAB (The MathWorks, Natick, MA). ',\n",
       " 'Tests of gender interactions with psychosocial work factors were performed using SAS software.',\n",
       " 'To test the relevance of our previously described signature ECM3 [4] in HER2-positive BCs, we studied the expression profile of ECM-related genes by LAS method in 5 cohorts of primary HER2-positive BCs. ',\n",
       " '[3] Analyses were performed using Stata 13.1 survey statistical methods that incorporated NHIS supplied probability weights, strata and primary sampling unit data. ',\n",
       " 'Data were extracted on paper files and then entered in an electronic database by data clerks and coded collected data were kept in an Excel database.',\n",
       " 'HPLC/MSn analysis was performed with an Agilent 1100 Series HPLC and LCQ Deca XPplus ion trap mass spectrometer (Thermo Finnigan, San Jose, CA, USA) equipped with electrospray ionization (ESI) source, with Xcalibur 1.3 controlling software. ',\n",
       " 'The Addressee moved the squirrel to the location deemed appropriate given the movements of the Communicator (event 3).',\n",
       " 'Availability and implementation: LCR-eXXXplorer is freely available at the URL http://repeat.biol.ucy.ac.cy/lcr-exxxplorer.\\n',\n",
       " 'Resulting trees were visualized inPhylodendron (http://iubio.bio.indiana.edu/soft/molbio/ java/apps/trees/).',\n",
       " 'The persistent estrus female rat is a model that is under constant estrogenic stimulation, mimicking polycystic ovarian syndrome. ',\n",
       " 'Studies were exported to remove the duplicates in Refworks™ and then exported to Microsoft Excel.',\n",
       " 'However, MAGMA still identifies a very large number of associated genes despite factoring in the LD information. ',\n",
       " 'The entire sequence of touches for each participant that contains the special runs was then entered into a Monte Carlo program (TouchStat 3.0) [41] to determine if they were likely to have occurred by chance.',\n",
       " 'Property developers generally make investment decisions based on the prices in the recent past period [69].',\n",
       " 'Structural Equation Modeling (SEM) was conducted using Stata 15.0 to assess the interrelationship between variables associated with lifetime physical IPV.',\n",
       " 'OAE was visualized with the Protégé 4.0.2 OWL editor.',\n",
       " 'SNPdetector was implemented in C and Perl.\\n',\n",
       " 'We also used mean reaction time on mixed trials minus mean reaction time on “hearts only” trials',\n",
       " 'For each voxel of the amygdala, the count ratio of fiber tracts reaching only the specific ROI to fiber tracts reaching any of the four ROIs [22, 23] was computed. ',\n",
       " '150 μl of the supernatant was evaporated to dryness under nitrogen flow at 50°C and reconstituted in 150 μl of a mixture of 0.1% formic acid and acetonitrile (90/10, v/v). ',\n",
       " 'Additionally, raw data collected during pre- and post-LiCl sessions were used in mixed model repeated measures analyses that tested for an interaction of session (pre or post LiCl) with LiCl treatment condition (paired or unpaired).\\n\\n',\n",
       " 'To determine the sources of small clusters other than these 11, we used representative sequences from every bacterial cluster containing more than 3 sequences as queries in blastn searches against GenBank. ',\n",
       " 'Study population.\\n',\n",
       " 'Here we present RNA-SeQC, a metrics tool with application to two domains: experiment design and process optimization; and quality control prior to computational analysis.\\n',\n",
       " 'Preprocessing was conducted using AFNI (NIH) and its associated functions, and most steps were identical to those listed in Berns, Brooks & Spivak (in press). ',\n",
       " 'Statistical analysis was carried out by SPSS 13.0 software and P<0.05 (two-tailed) was viewed as statistically significant.',\n",
       " 'We defined the surface searchlights [44] in PyMVPA [45](http://www.pymvpa.org) as cortical disks.',\n",
       " 'Data sets obtained in this study have been uploaded to GEO under accession ID GSE89392 (link for the reviewers: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?token=yrkpgamgzbcdlaz&acc=GSE89392).',\n",
       " 'The Ensembl data-mining tool BioMart (http://www.ensembl.org/index.html) was then used to convert these mouse gene IDs to human gene IDs, which resulted in a final set of 778 human genes. ',\n",
       " 'For CoNIFER-based CNV discovery, reads from each exome were split into up to two consecutive 36mers and mapped using the single-end mode of mrsFAST [75], then aligned to the hg19 reference genome. ',\n",
       " 'Statistical analyses were performed using SAS software, version 9.4 (SAS Institute, Cary, North Carolina). ',\n",
       " 'Second, multivariate logistic regression models were built to examine the independent relationship between PV and key demographic variables. ',\n",
       " 'The novelty of our complex HMMs is the introduction of more match states based on DaliLite structural alignments, which have aligned core blocks (structurally superimposable, shown in uppercase letters) and unaligned regions (structurally not superimposable, shown in lowercase letters)',\n",
       " 'Finally, Hierarchical Linear Modeling (HLM) with R was conducted to examine the fluctuations of perceived justice and the effect of the above-mentioned determinants on athletes’ perceived justice of the coach in a longitudinal perspective. ',\n",
       " 'All microarray data generated in this study are available at ArrayExpress (http://www.ebi.ac.uk/arrayexpress/ under identifiers E-MTAB-3306, E-MTAB-3307 and E-MTAB-3308).',\n",
       " \"We therefore explored the proper value range of F using simulated NGS data with sequencing qualities, which were generated using the tool 'Maq simulation' in the MAQ package (http://maq.sourceforge.net/).\",\n",
       " 'To demonstrate its capacity to optimize challenging pathways, we have applied Redirector to the production of myristoyl-CoA, examining the highest growth parameter design as well as the dependency analysis of multiple designs.\\n',\n",
       " 'Similarly, rural and urban populations may have differential access to various food types, possibly with seasonal variations. ',\n",
       " 'We then established the preliminary species assignment using NCBI’s BLAST search and we further confirmed this by aligning the sequences with in-house control sequences using the MEGA5 software [34].',\n",
       " 'The ESPRIT-Forest software is available at http://www.acsu.buffalo.edu/∼yijunsun/lab/ESPRIT-Forest.html, including the source code of both MPI and OpenMP implementations with pre-compiled executable codes.\\n',\n",
       " 'Field identifications were made based on morphological criteria according to [17], [29]. ',\n",
       " 'All statistical analyses were performed using SPSS (19.0).\\n\\n',\n",
       " 'The highest accuracy found for V against Immunoratio is also depicted in Figure [8].',\n",
       " 'Additional alignments to a transcript database ( Ensembl version 75, http://www.ensembl.org) (36) and the specific piRNA database piRNABank (http://pirnabank.ibab. ac.in/) were also performed (37).',\n",
       " 'Validation was performed by qRT-PCR for CXCL1 on individual samples from SC- and IV-infected mice at day 4 p.i. Specific primers for CXCL1 were designed using Oligo Calc (Kibbe, 2007) as follow: 5’- CACTGCACCCAAACCGAAGTCATA-3’; 5’-TCTCCGTTACTTGGGGACACCTTT -3’; A DNA Master Plus SYBR Green Kit (Roche Diagnostics, France) was used in a LightCycler 2.0 (Roche Diagnostics, France) with an initial incubation of 10 min at 95°C, 40 amplification cycles of ten seconds at 95°C, of 8 sec at 60°C, and of 10 sec at 72°C, during which the fluorescence data were collected. ',\n",
       " 'Our primary analysis workflow included three major steps as shown in Fig 1.',\n",
       " 'OpenEpi (Atlanta, GA) [25] was used to perform t-tests and χ2 tests of independence, and SAS version 9.2 (Cary, NC) was used for descriptive statistics and logistic regression modeling. ',\n",
       " 'We computed our PPI in AFNI as described in http://afni.nimh.nih.gov/sscc/gangc/CD-CorrAna.html.',\n",
       " 'Our pharmacokinetic modeling and simulation were performed by use of ADAPT II program [44] and a MATLAB program developed in our own lab for numerical solution of differential equations defined in Eqs.',\n",
       " 'Retention was assessed in two specific transfer cohorts of Microbiology and Cell Science majors who transferred into the university as juniors in the Fall of 2011 and 2012: on-campus transfers and distance education transfers.',\n",
       " 'Hierarchical Bayesian parameter estimation using Monte-Carlo Markov Chain was performed using Stan [50].',\n",
       " 'To analyze associations between individual and environmental variables and walking for transportation, we estimated several multilevel logit ordinal regression models using the Dual Quasi-Newton optimization estimation approach of Proc Glimmix in SAS 9.3 (SAS Institute Inc., Cary, North Carolina). ',\n",
       " 'Error bars for single-session behavioral data denote the 95% confidence-interval [16] implemented in the Python SciPy package.',\n",
       " 'One or two traps were placed in a room based on the areas, and five traps were placed in a household for 30 houses. ',\n",
       " 'Here we introduce CloVR-ITS, an automated pipeline for comparative analysis of internal transcribed spacer (ITS) pyrosequences amplified from metagenomic DNA isolates and representing fungal species.\\n',\n",
       " 'English transcripts were coded in Nvivo 9 and then analyzed by the main author, who was not involved with the study intervention. ',\n",
       " 'Statistical analysis was performed using SAS 9.3 (SAS Institute, Cary NC).\\n\\n',\n",
       " 'We implemented our efficient algorithm as software named ‘CapR’, which can compute the structural profiles for tens of thousands of long RNAs within a reasonable time by enumerating all the possible secondary structures of the RNAs.\\n',\n",
       " 'The interaction term was retained in the model if significant [57].',\n",
       " 'The software and documentation for M-Track are freely available for download from the Scimemi lab website (https://sites.google.com/site/scimemilab2013/software) or from the GitHub repository (https://github.com/scimemia/M-Track).\\n',\n",
       " 'To calculate ORs and SE from the ALSPAC BMI summary statistics we used genotype counts from SNPTEST output.',\n",
       " 'Missing values were replaced using multiple imputation implemented via the mice package [36] in R. Thirty imputations were computed and the pooled results across the imputed data sets are reported.',\n",
       " 'Max projections of the Z-stacks were used for primary cilium measurement and counting in ImageJ (NIH).',\n",
       " 'For RNAi knockdown, we designed pairs of single-stranded DNA oligonucleotides (one encoding target pre-miRNA and the other corresponding its complement) for 10 potential target sequences by using RNAi Designer (Invitrogen, Carlsbad, CA;https://rnaidesigner.invitrogen.com/rnaiexpress/).',\n",
       " 'All analyses were conducted using Stata version 13[41].\\n\\n',\n",
       " 'Our approach identifies regions like these by representing electrostatic isopotentials with volumetric solids generated by the new program VASP-E (Volumetric Analysis of Surface Properties with Electrostatics).\\n',\n",
       " 'In 2011 (T2), 2012 (T3), and 2013 (T4), 83, 46, and 43 additional high-risk HCWs newly joined the program, respectively, because of movement from other departments or new recruitment. ',\n",
       " 'Protein amounts were quantified from scanned blots using ImageJ (NIH).',\n",
       " 'To exclude these unspecific keywords, we eliminated the largest cluster of the nine cluster solution and re-ran the K-Means clustering on the remaining 602 maps. ',\n",
       " \"For each of these papers on PubMed Central, we retrieved its citations from PubMed Central, PubMed's “related citation” list, and selected search terms from each paper to run queries on Pubmed abstracts and titles using the “Entrez Utilities Entrez Programming Utilities” <http://eutils.ncbi.nlm.nih.gov/>). \",\n",
       " 'The automatic classification was performed in MathWorks Matlab, using the PRTools toolbox [57]. ',\n",
       " 'Statistical analyses were performed under a 5% significance level, using the GraphPad Prism v5.02 software.\\n\\n',\n",
       " 'NET framework (version 3.5) platform.',\n",
       " 'Data were entered in double by two data clerks using Epidata [49].\\n\\n',\n",
       " 'Praat version 5.3.32 for PC was used for all the simulations.',\n",
       " 'In the UCSC table Browser, MER sequences and their genomic regions were compared using RepeatMasker 3.3 (http://www.repeatmasker.org).',\n",
       " 'Strain variation was assessed using DNAsp package [29] to estimate average pairwise sequence divergence, minimum number of recombination events, and polymorphisms restricted to a single locality or shared between localities. ',\n",
       " 'Tests on initiatorship and overtaking were run with PASW Statistics 18 (SPSS Inc., Chicago, IL, 2009), the other tests were run using R software (R Development Core Team, Vienna, Austria, R2.11.1). ',\n",
       " 'All statistical analyses were performed using the statistical package SAS, version 9.1 (SAS Institute, Inc., Cary, North Carolina). ',\n",
       " 'Animal movements were recorded and digitally processed through the Optical flow algorithm based on the Horn-Schunck methodology [13] implemented in Matlab.',\n",
       " 'Therefore, the demand for industrial space can be represented as a function of economic activity in urban areas and regions. ',\n",
       " 'Stata version 12.0 SE was used for all analyses.',\n",
       " 'We have demonstrated that the absence of MCP-1 ameliorated chronic inflammation along with decreased fat mass induced by OVX.',\n",
       " 'PLINK (http:// pngu.mgh.harvard.edu/~purcell/plink) is a free, opensource whole-genome association analysis toolset, which efficiently performs association studies for single SNPs or computationally intensive genome-wide association studies.',\n",
       " 'Then, the hypotheses were tested with OLS Ordinary Least Square through the PROCESS macro for SPSS, which assesses the conditional indirect effects [38]. ',\n",
       " 'As preference for future living arrangement was a non-ordered categorical variable, a multinomial regression model was performed using SPSS Version 20.0 (IBM SPSS, Armonk, NY, USA) to estimate the associations between various attributes and preferences for future living arrangements. ',\n",
       " 'The analyses were performed using SPSS version 18 software.\\n\\n',\n",
       " 'All the data were analyzed using SPSS Statistics (SPSS Inc. Chicago, IL, USA). ',\n",
       " 'Data processing was performed by using the MultiPak software v. 9.8.0.19.\\n\\n',\n",
       " 'Statistically significant moderation paths (see Fig 2) were probed using PROCESS, an add-on for SPSS [32]. ',\n",
       " 'We used SAS 9.4 (SAS Institute, Cary, NC, USA) for data set-up and analyses.\\n\\n',\n",
       " 'Each surface-searchlight connectivity target has a radius of 13 mm and is centered on a node of a coarse surface with a total of 1284 nodes covering both hemispheres.',\n",
       " 'Stimuli consisted of faces in the center of the screen displaying either happy, sad, angry, or neutral emotions. ',\n",
       " \"Intracerebroventricular (ICV) injection of either vehicle (VEH; PBS, pH 7.4) or antisense was administered into the left hemisphere of adult rat's brain as described in supplementary information (Supplementary Information File S1). \",\n",
       " 'The analyses were conducted by using the PROCESS macro written by Andrew F. Hayes [38], which can be implemented in SPSS, and it is based on an ordinary least squares regression. ',\n",
       " 'Confirmatory factor analysis with Mplus version 7.3 [51] was used to estimate the degree of fit for each grooming measure (TTM, SP and NB). ',\n",
       " 'All statistical analyses aside from the CFA were performed using STATA version 11 [28].\\n\\n',\n",
       " 'Stimuli were controlled by a computer running Matlab (Mathworks) with Psychtoolbox [73] and Eyelink Toolbox [74].',\n",
       " 'Electrophysiological data were analysed using pClamp 10 and Igor pro software (WaveMetrics).',\n",
       " 'SgII was exclusively identified in SK-N-BE2 secretome (141-fold).',\n",
       " 'Results are publicly reported aggregated at whole trust level.\\n\\n',\n",
       " ').',\n",
       " 'Inductively Coupled Plasma Atomic Emission analyses were performed with an ICP-AES ThermoScientific iCap 6000 Series spectrometer (ICAP 6300 with dual view, empowered by iTeva software; Thermo Scientific, Waltham, MA, USA).\\n',\n",
       " 'Embryo transfer or caesarean derivation followed by barrier maintenance can be used to rear mice that are free of PVM.',\n",
       " 'The residual GM effect size (after age, weight, sex, and number of scans included was controlled) was calculated and correlated with social group size with nonparametric bivariate analyses in SPSS. ',\n",
       " 'A second major version of CellProfiler, rewritten in Python from its original MATLAB implementation, was published in 2011 [5] and included methods for tracking cells in movies and measuring neurons, worms, and tissue samples.\\n',\n",
       " 'The original images overlaid with the cell borders identified by PSICIC were manually verified to check for false positives.\\n',\n",
       " 'Data analyses were conducted using STATA version 14 (StataCorp, College Station, TX, USA). ',\n",
       " 'However, these protocols were used in rodent tissue and either did not require cultivation for longer than 3 days or did not monitor viability or functionality.\\n',\n",
       " 'Adapter removal was performed using the Trimmomatic utility v32.0 ( [52]), with a stringency parameter of 7, very short reads <8 nt were excluded. ',\n",
       " 'Wilcoxon rank-sum tests were undertaken in Stata 11.\\n\\n',\n",
       " 'Recommendations, defined as strong declarative statements [27] that were actionable and applicable by a multidisciplinary intersectoral team of professionals in a comprehensive hospital-based elder abuse intervention, as determined by the research team, were collected in a separate excel table. ',\n",
       " 'Relative gene expression was analysed with the Qbase+ software (Biogazelle).\\n\\n',\n",
       " 'Slatkin’s linearized FST pairwise genetic distance matrices between population [61] were calculated using the software Arlequin ver. 3.11.',\n",
       " 'To demonstrate how one can carry out Monte Carlo routines with our framework, we provide full source code listing for the C++ class that implements Stability Selection in our Supplementary Material. ',\n",
       " 'It is, however, known that marine microorganisms can express α-1,3-(3,6-anhydro)-L-galactosidase and α-1,3-(6-sulfate)-L-galactosidase, enzymes that catalyze the degradation of agar, the major component of the cell walls of red algae [88,89]. ',\n",
       " 'Statistical analyses were carried out using Stata 14.1.',\n",
       " 'The intensity was subsequently analyzed by using Nikon NIS Advanced Research software.\\n\\n',\n",
       " 'IHC profiler is currently compatible for use with Microsoft Windows operating system.\\n',\n",
       " 'The statistical analysis of the phenanthrene degradation data were performed by a parametric one-way ANOVA test using the SigmaPlot/SigmaStat software program (SPSS Inc., Chicago, Illinois, USA).\\n\\n',\n",
       " 'Sequences were aligned using VectorNTI 10.0 (Invitrogen Corporation, Carlsbad, USA) orCLC main workbench 5 (CLC bio, Aarhus, Denmark), and alignments were edited manually using the BioEdit software (http://www.mbio.ncsu.edu/bioedit/bioedit.html).',\n",
       " 'Data from physical examinations [including height, weight, and systolic and diastolic blood pressure (SBP and DBP, respectively)] and biochemical testing [total cholesterol (TCHO), total triglycerides (TG), low-density lipoprotein cholesterol, high-density lipoprotein cholesterol, fasting blood glucose (FBG), and glycated hemoglobin A1c (HbA1c) levels] were obtained from all participants. ',\n",
       " 'The ESA (European Space Agency) Globcover land cover dataset (http://due.esrin.esa.int/globcover) was used based on 22 land cover classes as per the UN-FAO Land Cover Classification System at 300-m spatial resolution [21].',\n",
       " 'Here we describe PyPhi, a Python software package that implements IIT’s framework for causal analysis and unfolds the full CES of discrete Markovian dynamical systems of binary elements.\\n',\n",
       " 'All such interviews were conducted by native language interviewers in each country using CATI-SPSS Dimensions software.\\n\\n',\n",
       " 'All statistical analyses were performed using SPSS 12.0.\\n\\n',\n",
       " 'A total of 21 optical sections each 10\\u2009µm apart were recorded from both mirrors (xymzt mode in the LAS X software) resulting in a volume of 735\\u2009×\\u2009735\\u2009×\\u2009200\\u2009µm.',\n",
       " 'For both the primary GLM and the PPI analysis, we then calculated the average smoothness of the residuals using 3dFWHMx and then used 3dClustsim to estimate the significance of different cluster sizes across the whole brain after correcting for familywise error (FWE).\\n\\n\\n',\n",
       " 'SAS version 9.2 (SAS Institute, Inc., Cary, NC, USA) was used for all the analyses. ',\n",
       " 'The CAD system was developed based on mathematical morphology using MathWorks.',\n",
       " 'To evaluate risk factors affecting diagnosing PBM by MRCP, a stepwise logistic regression analysis was performed with SAS 8. ',\n",
       " 'Figure identifier and caption: 10.1371/journal.pone.0192614.g002 Cities in the Beijing–Tianjin–Hebei region.This map was generated using ArcGIS 10.2.2 (www.esri.com).',\n",
       " 'READ was then applied to these data sets and the median of all average P0s per population was used to normalize scores assuming that this would represent an unrelated pair.',\n",
       " 'The analyses were run using STATA MP 11.2.',\n",
       " 'The program COLONY uses a maximum likelihood method to assign siblings and parentage using individual genotypes at all markers.',\n",
       " 'To meet these needs, I have developed ‘dcGOR’, a flexible R package that provides a basic infrastructure suitable for representing ontologies and annotations.\\n',\n",
       " 'All statistical calculations, except those pertaining to the Gower calculations, were made using SPSS 18 (IBM Corp., Somers, NY, USA).\\n\\n',\n",
       " 'These were classified into major taxonomic lineages using blastn against greengenes (greengenes.lbl.gov/) for prokaryotic sequences and against silva (www.arb-silva.de/) for eukaryotic sequences.',\n",
       " 'Results were analyzed in Stata (version 13.1).',\n",
       " 'Further data editing used SQL and SPSS software.',\n",
       " \"Statistica software (TIBCO Software Inc. Palo Alto, CA, USA) specifically uses Lloyd's method to implement the k-Means algorithm [38]. \",\n",
       " 'The package is distributed under GNU General Public License and available at http://sites.google.com/site/neuromapsoftware.\\n',\n",
       " 'SNPdetector runs on Unix/Linux platform and is available publicly (http://lpg.nci.nih.gov).\\n',\n",
       " 'All analyses were performed using SPSS (Version 22, release 22.0.0.2, IBM Corp., USA).\\n\\n',\n",
       " 'The CRISPRFinder web interface is accessible through http:// crispr.u-psud.fr/Server/CRISPRfinder.php',\n",
       " 'Data were cleaned in Access and then analyzed using SAS 9.3 (SAS Institute Inc.).\\n\\n',\n",
       " 'Custom Basic code was written to control the microscope.',\n",
       " 'In the lowland zone, 10 focus group discussion were conducted among the Ilchamus and 4 in a rural town that had mixed communities. ',\n",
       " 'Image analysis was performed as described earlier (Minin et al., 2006) using an open source image analysis software, ImageJ (http://rsbweb.nih.gov/ij).',\n",
       " 'DELLY is suitable for detecting copy-number variable deletion and tandem duplication events as well as balanced rearrangements such as inversions or reciprocal translocations.\\n',\n",
       " 'Acute medical admission units in all NHS Acute Trusts in England.\\n\\n',\n",
       " 'To compare the findings of the bivariable and multivariable analysis for each year with each other, a test of difference for comparison of two independent studies was used [17].',\n",
       " 'Their ages varied between 100 and 106 years, with a mean age of 102 years. ',\n",
       " 'The program can be freely downloaded from \"http://www4a.biotec.or.th/GI/tools/metasel\".\\n',\n",
       " 'This group of viruses usually causes asymptomatic infections of the intestinal tract.',\n",
       " 'Preliminary quality control of raw reads was carried out with FastQC software v0.11.2 [89].',\n",
       " 'Analysis of the data occurred using SPSS.',\n",
       " 'The source code of LAILAPS-QSM is available under GNU General Public License version 2 in Bitbucket GIT repository: https://bitbucket.org/ipk_bit_team/bioescorte-suggestion\\n',\n",
       " 'A distribution of averages was then used to derive 95% confidence intervals using the first-order normal approximation as implemented in the boot package for R [114].\\n\\n',\n",
       " 'The beta-version of \"OnlineTED\" was made available on November 1 st , 2012 at http://www.OnlineTED.com (German version: http://www.OnlineTED.de).',\n",
       " 'All analyses were performed using SPSS for Windows version 23 software (IBM Japan Ltd, Tokyo, Japan).\\n\\n',\n",
       " 'If the transfer is not successful, usually due to network problems or by the user cancelling the transfer, the sequence is not updated and this information is reported back to the ReMIT Client, so that the user knows that they need to transfer the image again. ',\n",
       " 'For the second dataset, phylogenetic relationships were also inferred using Bayesian inference as implemented in MrBayes v.3.1.2 [49]. ',\n",
       " 'In formulating the model, we focused on interactions among AMPK, MTORC1, and ULK1, because this triad of kinases is recognized as playing a critical role in regulation of autophagy and translation [50]. ',\n",
       " 'The MAGPIE source code can be downloaded from the GitHub repository at https://github.com/christbald/magpie and installed on any public or private server, as well as on any local computer.\\n',\n",
       " 'These differences were extremely significant in some populations and impacted the final yields.Wheat seedling characteristics of different grades of cultivated land.\\n',\n",
       " 'Thus, the canonical model describing the mode dynamics employs the nonlinear rate equations:(1)where Ai≥0, i\\u200a=\\u200a1,…,N, represents the cognitive modes, Bi, i\\u200a=\\u200a1,…,M, represents the emotional modes and Ri, i\\u200a=\\u200a1,…,K, represents the resources consumed by these mental processes. ',\n",
       " 'Results involving these variables should therefore be interpreted as exploratory.',\n",
       " 'Data from the surveys were double-entered and analyzed using SPSS ver. 19 (IBM, Armonk, NY, USA).',\n",
       " 'The SRM data were analyzed using the Skyline software [18].',\n",
       " 'By contrast, with the gait experiments performed before and after 2 months of supplementation, there is limited stress and rats were encouraged to move freely across the walkway of the GaitLab system. ',\n",
       " 'Data from GeneChip microarrays has been deposited in the NCBI Gene Expression Omnibus (GEO) and is accessible through the GEO Series accession number GSE88723 (available prior to publication under: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?token= wtgfgkwczzgtfob&acc= GSE88723).',\n",
       " 'Data were coded using MAXQDA 2007 qualitative data analysis software [37] and were analyzed both inductively and deductively.',\n",
       " 'All analyses were performed using Stata 12.0 SE.\\n\\n',\n",
       " 'The VideoTrack system calculated the time, the distance, and the velocity of exploration within the full quadrant, and within a virtual central zone of 2,921 cm2 (Fig.',\n",
       " 'VIP uses a phylogenetic tree (ML tree of the combined dataset) and direct geographic information data to search for the disjunctions/barriers among sister groups.\\n\\n',\n",
       " 'All three images (DWI, FA and MD) were masked using a mask generated from the registered FreeSurfer image, including cortex, white matter and subcortical regions. ',\n",
       " 'After addition of each concentration of Mch (10 to 10\\xa0M), pictures were recorded in 5\\xa0s intervals for 3\\xa0min with a stereo microscope (Discovery V8; Zeiss, Jena, Germany) controlled by the Axio Vision 4.8.2. ',\n",
       " 'Other murid herpesviruses\\n',\n",
       " 'All statistical analyses were performed in SPSS 23.\\n\\n',\n",
       " 'Multiple autoradiographs were obtained using different exposure time, and immunoreactive bands within the linear range of detection were quantified by densitometric scanning using Scion image software (Scion Corp., Frederick, MD).\\n\\n',\n",
       " 'A posteriori coil reconstruction of the coil position was based on custom-written MATLAB (MathWorks Inc, MA, USA) scripts. ',\n",
       " 'The original pictures were manipulated using Adobe Photoshop software (Adobe Systems; San Jose, CA, USA) and Pixelmator (Pixelmator Team Ltd.; Vilnius, Lithuania). ',\n",
       " 'The tool is built in MS Excel 2010 and consists of two modules:',\n",
       " 'The Altmetric bookmarklet was installed in the Firefox browser bookmarks toolbar.',\n",
       " 'We used CLUMPP 1.1 [59] to average the results of multiple iterations for a given K. GENELAND 3.3.0 [60] was used to perform a spatial genetic analysis by integrating geographic and genetic information and to determine the most probable K. Ten runs of 1×105 MCMC iterations were performed for K from 1 to 28 with a spatial coordinates uncertainty of 1 km.',\n",
       " 'The TIS score was analysed using the multilevel model framework as outlined above for the primary outcome (not transformed).',\n",
       " '[22].',\n",
       " 'The participants were instructed to avoid consuming liquids after crossing the finish line until completion of the measurements.',\n",
       " 'For richness and evenness estimates, EstimateS [14] was used. ',\n",
       " 'Mobyle is a generic web-based framework.',\n",
       " 'All analyses were conducted using SAS Version 9.4 using survey procedures (PROC SURVEY REG) that account for the complex survey design.',\n",
       " 'We present an open source fragmentation methodology, an accompanying command line tool and a corresponding web service called FragIt that enables one to easily fragment any molecule or system of interest using predefined (or custom) patterns to locate fragmentation points.\\n',\n",
       " 'Statistical analyses were performed using SPSS version 19.0.',\n",
       " 'The complete source code tree, example files, and documentation for IDEPI; for the most current version visit the package distribution page at https://github.com/veg/idepi.\\n',\n",
       " 'We used EstimateS v 8.2.0 [11] to generate a presence-absence accumulation function of subtypes and calculated the nonparametric estimate of subtype richness with 95% confidence intervals using the Chao2 estimate and 50 randomizations with replacement [6].',\n",
       " 'The ORTEP (Oak Ride Thermal Ellipsoid Plot Program) of compound 7 and 9 are shown in Figure 2 and X-ray crystallographic data presented in Table 3.\\n\\n',\n",
       " 'The sequence data is run through filtering and analysis pipelines, such as QIIME (qiime.org), in order to define operational taxonomic units and is compared to bacterial 16SrRNA databases (such as Green Genes, http://greengenes. lbl.gov/cgi-bin/nph-index.cgi) to determine the bacterial diversity of the sample.',\n",
       " 'In the RSS data, after registration and down-sampling of the FreeSurfer atlas to the fMRI-space, in two regions of one subject no voxels survived the down-sampling (the left entorhinal cortex and the right temporal pole). ',\n",
       " 'For excitation of intracellular NADH a 8 MHz pulsed frequency-tripled Nd:vanadate laser tuned at 355 nm (HighQ Laser, Austria) was used. ',\n",
       " 'The analysis took about 2.5 hours to run in a Windows 7 Professional desktop computer with 8-core i7-3770 3.40 Ghz CPU and 8 Gb RAM.',\n",
       " 'Subjects performed a version of the ‘Cyberball’ social exclusion task whilst being scanned [19].',\n",
       " 'We further BLAST searched this transcriptome for sequences with homology to the viruses genomes derived from our transcriptome data.\\n\\n',\n",
       " 'For neuroimaging data pre-processing we used standard FMRIB Software Library (FSL) tools v5.0 (http://fsl.fmrib.ox.ac.uk/fsl/fslwiki/) [21].',\n",
       " 'All the analyses in this paper were performed with STATA 11.0 software.',\n",
       " 'To rationalize the anti-cancer potential of FKB derivatives, the online web Prediction of Activity Spectra for Substances (PASS) server was used to predict the biological activity of potential FKB derivatives [49,50]. ',\n",
       " 'Data analyses were performed using R software (R Development Core Team, 2010, version 2.11.1). ',\n",
       " 'VisANT is available at http://visant.bu.edu.\\n',\n",
       " 'These analyses were performed using IBM SPSS v. 21.\\n\\n',\n",
       " 'To infer ancestral areas of L. neilli populations and investigate the relative role of vicariance and dispersal in the evolutionary history of this species, we used a dispersal–extinction–cladogenesis (DEC) model of range evolution implemented in LAGRANGE [64]. ',\n",
       " 'Analyses were conducted using SPSS version 20.0.',\n",
       " 'The tank used for the MWM was divided into four identical quadrants (6,358 cm2 each) using two 180 cm long opaque plastic separators.',\n",
       " 'Based on a structural support vector machine (SSVM) model, PhylopythiaS permits taxonomic assignment by analyzing the compositional vectors derived from DNA sequence fragments. ',\n",
       " 'We used a multiple imputation with k = 10 imputed data sets via the SAS procedure MI.',\n",
       " 'A force-directed algorithm with consideration of weights linearly proportional to our measure of revealed relatedness appeared to produce the most effective network layout (for description of the force-directed layout used in Cytoscape software see http://cytoscapeweb.cytoscape.org/documentation/layout).',\n",
       " 'VDJtools can calculate basic immune repertoire statistics that were commonly used in pre-HTS era repertoire analysis.\\n',\n",
       " 'During mitosis, the microtubule (MT) cytoskeleton reorganizes to assemble the mitotic spindle that will capture chromosomes via their kinetochores and segregate them equally into two daughter cells. ',\n",
       " 'The workflow between the ReMIT Client app, the web-based interface and the PACS server is shown in Figure 2.\\n\\n',\n",
       " 'The swimming path of the mice was recorded by a video camera and analyzed by the Videomot 2 software (TSE).',\n",
       " 'The Figure was generated using GraphPad Prism version 7.01, GraphPad Software, La Jolla California USA, www.graphpad.com.\\n\\n\\n',\n",
       " 'All statistical analyses were performed with SAS software version 9.3 (SAS Institute Inc, Cary, NC).\\n\\n',\n",
       " 'Data are available from the St John of God Human Research Ethics Committee (contact via ethics@sjog.org.au) for researchers who meet the criteria for access to confidential data.',\n",
       " 'In breeding colonies, mice are infected shortly after weaning as maternal antibody levels wane. ',\n",
       " 'However, there is currently not a way of obtaining an overall p-value for categorical variables in SPSS, which is necessary for selective backward logistic regression. ',\n",
       " 'PhyloGibbs-MP is available at http://www.imsc.res.in/∼rsidd/phylogibbs-mp/.\\n',\n",
       " 'In an effort to address these needs, we have developed BEDTools, a fast and flexible suite of utilities for common operations on genomic features.\\n',\n",
       " 'We performed all analyses using Program R [31].',\n",
       " 'G*Power [47] was used to calculate sample sizes to detect a mode bias of 1.5 in population prevalence using McNemar’s Exact Test for paired nominal data, assuming 25% of participants would switch indicator status (above or below threshold) between rounds. ',\n",
       " 'A public release of DCE@urLAB, together with the open source code and sample datasets, is available at http://www.die.upm.es/im/archives/DCEurLAB/ and in Additional files [1] and [2].\\n',\n",
       " 'The web service [44] available at http://www.fragit.org enables users to upload their structure, fragment it and download the resulting input file to GAMESS.\\n',\n",
       " 'In response, we have created Wham (Whole-genome Alignment Metrics) to provide a single, integrated framework for both structural variant calling and association testing, thereby bypassing many of the difficulties that currently frustrate attempts to employ SVs in association testing.\\n',\n",
       " 'Uncorrected or under-corrected refractive error was determined to be the main cause of vision loss if the distance visual acuity in one or both eyes improved to better than or equal to 6/12 (≥6/12) with pinhole testing or auto-refraction.',\n",
       " 'Predictions of surface patches for the representative domain structures were determined from the SHARP2 server [46].',\n",
       " 'In the current implementation, DOOR uses the Àlogarithm of BLAST E-values or alignment bit-scores to measure the level of similarity between two operons.',\n",
       " 'As part of the laboratory protocol, eyes-closed measurements were collected at the beginning of each experiment before testing. ',\n",
       " 'taveRNA involves web interfaces to the algorithmic tools we have developed for each one of the above problems: (1) alteRNA, an alternative thermodynamic model-based RNA secondary structure prediction program, (2) inteRNA, a program for predicting the joint secondary structure of two interacting RNAs and (3) pRuNA, a tool for quickly identifying all potential regulatory ncRNAs for a query mRNA.',\n",
       " 'Participants were originally recruited in the year 2000 for the Society, Environment and Reproduction (SER) study, a prospective, naturalistic, longitudinal study that focused on the relationship between “real life” daily stress and women’s reproductive function [27, 28]. ',\n",
       " 'The multiple mediation analysis was performed using PROCESS for SPSS [29].',\n",
       " 'We used IBM SPSS 22.0 for the quantitative analysis.',\n",
       " 'To evaluate READ’s performance, we calculate false positive and false negative rates. ',\n",
       " 'From each selected AWC area, four eligible households were identified through systematic random sampling. ',\n",
       " 'We performed GSA using MAGMA [26](http://ctg.cncr.nl/software/magma) and summary statistics from various GWAS.',\n",
       " 'Alignment result files were parsed using the R statistical language (http://www. R-project.org) and the Bioconductor project (http://www. bioconductor.org) (38) to generate tables including the read count and annotation information for each entry in the target database.',\n",
       " 'Gaze Data Collection and Analysis: EyeVision data processing software was used to calibrate POG from the recorded data and the data stream for each of the ten riders was recorded from the DVCR using EyeVision for identification of POG.',\n",
       " 'The redundant sequences were removed and the sequences were aligned using SILVA SINA (www.arb-silva.de/aligner/) with the reference dataset SILVA SSU Ref NR [17].',\n",
       " 'Data on clinical trials conducted in India obtained from ClinicalTrials.gov is made available in the corresponding Supporting Information file.',\n",
       " 'We used a previously published model [7] as a basis for our mathematical model of rabies transmission incorporated into RabiesEcon (for equations, see S2 Appendix). ',\n",
       " 'All statistical analyses were done in Stata 12 (Stata Corp. Inc. TX, USA).\\n\\n',\n",
       " 'Data was entered and analyzed using Statistical Package for Social Sciences version- 16 (SPSS IBM, New York, NY, USA). ',\n",
       " 'Descriptive statistics of the outcome measure, psychological distress, and the percentage distribution of the variables in the sample were calculated using SPSS Complex Samples, weighting for gender, age and city district. ',\n",
       " 'Specifically, we used 3D Fast Fourier Transform (FFT), as implemented by the function ‘fftn’ in MATLAB, to convert the intact movie into the frequency domain, in which the phase was randomly shuffled while the magnitude remained unchanged; then inverse FFT was applied to the phase-shuffled data to generate a scrambled version of the original movie. ',\n",
       " 'Analogous to GRAB [14], we partition the genome in non-overlapping windows of 1 Mbps and calculate the proportions of haploid mismatches and matches, P0 and P1, for each window. ',\n",
       " 'NeuroMap is made freely available to the community under GNU General Public License (GPL) at http://sites.google.com/site/neuromapsoftware.\\n',\n",
       " 'We introduce RNA-SeQC, a program which provides key measures of data quality.\\n',\n",
       " '\"Project name: RankProdIt Project home page: http://strep-microarray.sbs.surrey.ac.uk/RankProducts/ Any restrictions to use by non-academics: None\"\\n',\n",
       " 'Protein identification, quantification, and phosphosite determination: All MS data was interpreted using the using the Spectrum Mill software package v4.0 beta (Agilent Technologies, Santa Clara, CA). ',\n",
       " 'All analyses were adjusted for the survey design, clustering of households and sample weights using STATA 11. ',\n",
       " 'Digitized retinal whole-mount reconstructions were examined frame by frame at high magnification, and every RT97 RGC was marked with a colored dot to identify either faint or strong RT97 immunoreactivity with the aid of Photoshop® software. ',\n",
       " 'Based on these two types of dendrimer topology, the concept of an “Isolation Chromophore” [23] and the “Ar-ArF” self-assembly [24] effect were subsequently utilized to achieve better performance (Schemes S2-S4†).',\n",
       " 'Doppler scans were replayed back on the quantification software (Q-LAB v6, Philips Healthcare, Andover, MA, USA) and three images per ovary with the greatest area of blood flow were chosen for quantification by two observers blinded to the treatment groups. ',\n",
       " 'Out of a total of 1,342 RT97 RGCs found in the experimental retinas, 27% were strongly labeled and 73% were faintly stained. ',\n",
       " 'All statistical analyses were conducted using the computer program SPSS (version 18.0, SPSS, Inc., Chicago, IL).\\n\\n',\n",
       " 'To address these limitations we have developed Molecular Interactions in Super-Resolution (MIiSR), a validated set of software tools for quantifying intermolecular interactions and the formation of higher-order molecular complexes in super-resolution images.\\n',\n",
       " 'Finally, caregivers were counseled on recognizing symptoms of common childhood diseases (fever, diarrhea, acute respiratory infection, and so on), how to address these adequately, and how to recognize dangerous signs. ',\n",
       " 'We therefore, to make it easier to facilitate comparisons between no vaccination and dog vaccination programs, programmed into RabiesEcon a process to calculate a “steady state” of a near-constant number of annual cases of canine rabies in a “no vaccination” scenario. ',\n",
       " 'Some of them had been used diagnostically before were found in the European Best Practice Guidelines for Molecular Analysis in Spinal Muscular Atrophy, others were found using the Santa Cruz genome browser. ',\n",
       " 'Protein motifs were predicted using SMART (http://smart.emblheidelberg.de), and SignalP 3.0 was used to identify the signal peptide (http://www.cbs.dtu.dk/services/SignalP). ',\n",
       " 'Finally, the raw voltages, spectral densities, inferences of parahippocamapal activity, and microstate parameters were exported into a singular SPSS dataset for further analyses.\\n\\n',\n",
       " 'ImageJ [17] is a public domain, Java-based image-processing program developed at the National Institutes of Health.\\n\\n',\n",
       " 'When geo-referenced localities were not available (only locality names), geographic coordinates were obtained with the software Google Earth. ',\n",
       " 'h. Thereafter, lungs were removed from the fixative and placed in 70% alcohol for 2–7 days before paraffin embedding. ',\n",
       " 'Stata 11.1 was used for statistical analysis to calculate means, proportions and regressions. ',\n",
       " 'Genotypes with a read depth of <10 and GQ of <20 were converted to missing by using BCFtools [28].',\n",
       " 'The coronal SSFSE T2-weighted MR images were obtained in two or more breath-holds.',\n",
       " 'Whereas other methods provide an inflexible set of variant filters or prioritization methods, GEMINI allows researchers to compose complex queries based on sample genotypes, inheritance patterns, and both pre-installed and custom genome annotations.\\n',\n",
       " 'We developed a novel, integrated algorithmic approach (DriverNet) to analyze population-based genomic and transcriptomic interrogations of tumor (sub)types for identification of pathogenic driver mutations.\\n',\n",
       " 'Most of the data needed is included in the published article.',\n",
       " \"From the intrinsic perspective of the system, the amount of integrated information specified by a mechanism in a state cannot be more than either the cause or the effect integrated information, so the minimum of the two is taken [15]: Finally, again from the intrinsic perspective, the mechanism's causal role within the system in its current state can only be a single one—corresponding to the cause-effect repertoire that is maximally irreducible (a mechanism cannot perform multiple input-output functions over an overlapping set of elements, [15]). \",\n",
       " 'The performance of eMatchSite is evaluated using several datasets and compared to other algorithms for binding site matching in large-scale benchmarking calculations.\\n',\n",
       " 'The survey forms were scanned using an optical character recognizing (OCR) machine and data were analyzed using STATA 10.1 for Windows (Stata Corp, College Station, TX, USA) software. ',\n",
       " 'The journal is hosted on an open-access, open-review, rapid publication platform called PubPub, created by students at the Media Lab.',\n",
       " 'Follow-up in-person interviews at two-year intervals were conducted by trained physicians following the same procedures used in the baseline interview.',\n",
       " 'All analyses were conducted in the program R version 3.2.2 [57], with add-on packages ‘nlme’ [58] for mixed effects models.\\n\\n',\n",
       " 'Glyoxylic acid reacts with catecholamines in a gaseous reaction to form fluorescent products.',\n",
       " 'Fifty micrograms of the protein extractions were separated by 10% SDS-PAGE transferred to 0.22\\xa0mm nitrocellulose (NC) membranes (Sigma-Aldrich) and incubated with specific antibodies.The autoradiograms were quantified by densitometry (Quantity One software; Anti-p21 was from Abcam.',\n",
       " 'Since ggsashimi uses the most popular file formats and has very few dependencies, it can be easily integrated in any splicing analysis pipeline, and can facilitate the interrogation of alternative splicing in large-scale RNA sequencing projects, such as ENCODE [6] and GTEx [7]. ',\n",
       " 'Gene lists and log2 expression fold-changes were uploaded into the IPA tool (Ingenuity Systems, www.ingenuity.com).',\n",
       " 'ggsashimi is freely available at https://github.com/guigolab/ggsashimi.\\n',\n",
       " 'All processing was performed using ScopeWin and Matlab software.\\n\\n',\n",
       " 'Significant differences were determined using the Mann-Whitney non-parametric two-tailed test using GraphPad Prism Version 5.',\n",
       " 'This enabled automatic importation of entire interviews with codes into the qualitative data analysis software MAXQDA 10 (VERBI Software, Consult. Sozialforschung. GmbH, Marburg, Germany). ',\n",
       " 'Since April 1, 1968, all citizens in Denmark have been registered in the Civil Registration System. ',\n",
       " \"In order to use standard instrumental variable estimation methods (‘two-stage least squares') [17] the corr2data command in Stata was used to construct an artificial dataset matching the reported number of participants with each genotype, and the means and standard deviations of alcohol intake and blood pressure within each genotype. \",\n",
       " 'We present a bioinformatics solution (3PD) to support the experimentalist in detecting long-ranging intra or inter chromosomal contacts by Chromosome conformation capture (3C) assays.\\n',\n",
       " 'The statistical analyses were performed using SPSS software (version19.0; IBM Corp, New York).\\n\\n',\n",
       " 'For each patient the following information was collected from their medical records and analysed: Epidemiological variables: age, gender, state of residence in Nigeria, person who referred them to the CDTUB and delay in seeking treatment.Clinical variables: clinical form of the lesion (nodule, plaque, oedema or ulcer), location of the lesion, WHO classification based on category of lesions, restriction or not of joint movement.Biological variables were obtained through IS2404-PCR results from the Reference Laboratory for Mycobacteria in Cotonou on swab specimens (for ulcerative lesions) or on fine-needle aspiration samples (for non-ulcerative lesions).',\n",
       " 'The Statistical Analysis System for Windows, version 9.4 (SAS Institute, Cary, NC, USA), was used for analysis.\\n\\n\\n',\n",
       " 'Relatedness within populations was examined using three programs: MLRELATE [16], COLONY [17] and COANCESTRY [18]. ',\n",
       " 'MAGERI is implemented in Java v 1.8 and is distributed as a single cross-platform executable JAR file [https://github.com/mikessh/mageri].\\n',\n",
       " 'Gray matter volume (GMV) calculation: The GMV of each voxel was calculated using Statistical Parametric Mapping software (SPM8; http://www.fil.ion.ucl.ac.uk/spm/software/spm8/).',\n",
       " 'Woody debris at the base of the core prevented the recovery of sediments below 7 m depth.',\n",
       " 'With Pep2Path, we introduce a software package to fully automate the peptidogenomics approach through the rapid Bayesian probabilistic matching of mass spectra to their corresponding biosynthetic gene clusters.\\n',\n",
       " 'A total of eight cancers (hepatocellular carcinoma, thyroid cancer, colorectal cancer, gastric cancer, lung cancer, prostate cancer, breast cancer, and cervical cancer) were included in the present study. ',\n",
       " 'We present this method in the C4 online supplement and we used the APCD module (available on Stata) with our data in an attempt to reproduce their results.',\n",
       " 'Statistical analyses were performed using SPSS (version 17.0; SPSS Japan Inc., Tokyo, Japan).\\n\\n',\n",
       " 'Figure identifier and caption: 10.1371/journal.pone.0127450.g003 Ixodes scapularis nymph questing on stem in experimental arena.1 cm of a 10 cm dowel is visible.',\n",
       " 'Spectra were divided into 0.001\\u2009ppm-wide buckets over the chemical shift range (-0.2; 9.5\\u2009ppm) using the AMIX software (Bruker GmbH).\\n\\n',\n",
       " 'In order to determine the unique protein-specific tryptic sequences BLASTp analysis (http://blast.ncbi.nlm.nih.gov) was carried out searching the NCBI non-redundant protein sequence database.',\n",
       " ') ⊗ w(e\\n',\n",
       " 'Data were analyzed by using SPSS 17.0 for Windows XP (SPSS Inc., Chicago, IL, USA).\\n\\n',\n",
       " 'All statistical analyses were performed using STATA® version 10.0 and R Studio version 3.2.2 for Macintosh®.\\n\\n',\n",
       " 'These results suggest that VASP-E should prove a useful tool for the characterization of specific binding and the engineering of binding preferences in proteins.\\n',\n",
       " 'Two arenas were not seeded with nymphs and served as experimental controls for the effectiveness of the arena barrier and leaf litter heat treatment.',\n",
       " 'All statistical analysis was performed using the Stata/SE software (release 13.1) (Stata Corp LP, College Station, TX, USA), and the statistical significance level was set at P = 0.05 in all statistical inferences.\\n\\n',\n",
       " 'There is great interest among teachers in the potential of neuroscience to enhance educational practice (Goswami 2006; Pickering and Howard-Jones 2007), but considerable debate about the extent to which this potential can be realised (Varma et al. 2008). ',\n",
       " 'All reported P-values are based on two-sided tests and compared to a significance level of 5%.',\n",
       " 'Files were formatted for MCMC linkage analysis and an ideal set of SNPs was chosen for a marker panel with the Pedigree-Based Analysis Pipeline (PBAP) [68], targeting marker spacing of 0.5 centimorgan (cM), minor allele frequency (MAF) > 0.2, and LD between markers < 0.04. ',\n",
       " 'Gender and age had significant effects on experience scores for feeling treated with respect and dignity and for pain control, with younger age and female gender associated with worse reported experience. ',\n",
       " 'CT scans (≤1 mm slices) were obtained (DICOM file) for volumetric conversion to 3D models (STL file) using free open-source software (3D Slicer, http://www.slicer.org).',\n",
       " 'The duration of clinical and biological abnormality is 35 days on average in the severe AEs with systemic involvement [7].\\n\\n',\n",
       " 'Validated pQCT analyses were available for the Gothenburg and Malmö cohorts.',\n",
       " 'All data is available in the supporting files.',\n",
       " 'Collected eyelid data were quantified, through a purpose-designed Excel worksheet, as the percentage of CRs per session — namely, the proportion of stimulations within a session of 60 presentations that generated an EMG activity satisfying the above-mentioned criteria. ',\n",
       " 'Demographic, clinical, and laboratory data from the electronic medical record and study forms were de-identified, entered into an Excel spreadsheet, and exported into Stata v14 software (StataCorp, 2011, College Station, Texas) for analysis. ',\n",
       " 'All the statistical analyses were carried out with IBM SPSS Statistics 19.',\n",
       " 'Data processing, visualization, and random forest development were conducted using pre-existing libraries, particularly the randomForest package33 and ggplot2 package37 (for density plots and stacked bar plots) in the R programming environment, whereas spatial mapping was conducted in ArcMap 10.7.1, as indicated in the methods and validation.',\n",
       " 'The ROI was defined for each subject based on the peak coordinates of the results from the group level.',\n",
       " 'MAGMA was chosen because it corrects for LD, gene size and gene density (potential confounders) and has significantly more power than other GSA tools [68]. ',\n",
       " 'Finally, the traces were filtered with a loess filter (time window: 0.5 sec) using the ‘‘smooth()” function of Matlab, version R2013b. ',\n",
       " 'In immunocompetent mice in which infection is effectively cleared, a breeding suspension strategy for 8-10\\xa0weeks combined with excellent sanitation, filter tops and conscientious serological testing of offspring and sentinel mice has also been reported to be effective, and prolongation of breeding cessation up to 12\\xa0weeks resolved infection even in immunocompromised mice [393].\\n',\n",
       " 'Since heat stress was shown to increase the tonic immobility duration in laying hens [33], which could indicate a state of stress in animals, all the hens were tested once before the heat period and after 3 weeks of treatment. ',\n",
       " 'Further analyses regarding behavioural data (t-test for paired samples) were carried out using SPSS 11 (SPSS Inc., Chicago, USA).\\n\\n',\n",
       " 'ASPicDB is an ongoing project and we plan to further develop it in the next releases.',\n",
       " 'All statistical modeling and analyses were conducted using PASW Statistics 18, Release Version 18.0.0 (SPSS, Inc., 2009, Chicago, IL, www.spss.com). ',\n",
       " 'Here we introduce Bamgineer, a tool to modify existing BAM files to precisely model allele-specific and haplotype-phased CNVs (Fig 1).\\n',\n",
       " 'MAP was calculated using the values of systolic blood pressure (SBP) and diastolic blood pressure (DBP) by the Iox 2.9.5.73 software (Emka Technologies, Paris, France). ',\n",
       " 'I introduce an open-source R package ‘dcGOR’ to provide the bioinformatics community with the ease to analyse ontologies and protein domain annotations, particularly those in the dcGO database.\\n',\n",
       " 'All computations were done using SAS V9.2.',\n",
       " 'The reservoir neuron properties and synaptic connectivities were set almost identically to the network described in [63], and our simulation code incorporated MATLAB code from that work. ',\n",
       " 'All data were analyzed by SPSS (Version 23) by univariate and multivariate procedures, using two-sided t-tests and Chi2, respectively. ',\n",
       " 'We developed a Perl program to store all the graph paths for a given protein.',\n",
       " 'All registration steps were done using trilinear interpolation as implemented in FLIRT (Jenkinson and Smith 2001; FSL 4.1.4).',\n",
       " 'A fixation was defined by the BeGaze software as a stable gaze (within 0.8 visual degrees) for at least 60 ms. For each valid trial, we calculated the gaze arrival time by subtracting the time when infants first looked inside the Target AOI from the hand-object contact time (i.e., the end of the movement phase). ',\n",
       " 'Fish were filmed for 15 minutes and the behaviour of each sex was analyzed using JWatcher software, which calculates the total time and frequency of each quantified behaviour during this period. ',\n",
       " 'WGBS Blueprint data of B cells are available from the EGA under the accessions EGAD00001001590, EGAD00001001587, EGAD00001001548, and EGAD00001001160.',\n",
       " 'GeoTIFF dataset for different general circulation models are openly available via Figshare repository, https://figshare.com/s/6b18c6ce273a3ecaaddc.',\n",
       " 'For nonsyndromic cases, parental characteristics, case characteristics, and parent-reported neurodevelopmental outcomes were described using counts and frequencies for discrete variables, and means and standard deviations or median and range for continuous variables. ',\n",
       " 'All analyses were performed with multiple imputation methods using STATA version 14 (Stata Corp LP, College Station, TX, USA).\\n\\n',\n",
       " 'Resting-state fMRI data pre-processing consisted of motion correction, brain extraction, high-pass temporal filtering with a cut-off of 100 s, and field-map correction and was carried out using FSL Multivariate Exploratory Linear Optimized Decomposition into Independent Components (MELODIC) [25]. ',\n",
       " 'The 10 μL reaction mixture contained 1X DNA MasterPlus SYBR Green (QIAGEN, France), 0.5 μM of each primer, and 5 μL of template. ',\n",
       " 'For illustration, we plotted these FWE corrected activation maps onto the canonical Montreal Neuroimaging Institute (MNI) 152 template brain image 33 using SPM 8, and onto a three-dimensional-rendered template brain image using MRI- cron software (http://www.sph.sc.edu/comd/rorden/mricron/).',\n",
       " 'The data that support the findings of this study are available from the corresponding author upon reasonable request.',\n",
       " 'Data was analyzed by SPSS 16.',\n",
       " 'Off-line EEG processing and analyses were performed by adopting custom MATLAB (MathWorks) scripts using functions from the EEGLAB environment [38].',\n",
       " 'Subsequently, the clinical conditions deteriorated further, and the woman was admitted on an emergency basis with high fever, respiratory and cardiocirculatory failure, diffuse rash, generalized edema, and lymphadenopathy. ',\n",
       " 'Proteins were transferred in Tris-glycine buffer (pH 8.3) containing 25 mM Tris and 192 mM glycine and 20% methanol, for 1.75 h at 600 mA. ',\n",
       " 'Multistage sampling technique was used to recruit the respondents.',\n",
       " 'The dependent eye movement measures of total dwell time, relative dwell time, entry time, number of fixations, first fixation duration and average fixation duration were each subjected to a generalized linear mixed model analysis (GLMM; GENLINMIXED in SPSS) to account for repeated measures of the same subject. ',\n",
       " 'The MAFCO compression tool can be found at: http://bioinformatics.ua.pt/software/mafco; https://github.com/lumiratos/mafco.\\n',\n",
       " \"We have found a way to double the accuracy of Matlab's Ricatti equation solver lyap.m by essentially applying it twice.\",\n",
       " 'Participants had a short training session with Cyberball before playing the task in the scanner.',\n",
       " 'Spectral variables:Center of gravity refers to the spectral centroid, which accounts for the weighting of noise across the sample (log-transformed). ',\n",
       " 'Data were analyzed using SAS version 9.4 (SAS Institute, Carey, NC). ',\n",
       " 'The results showed an average correlation coefficient of R = 0.93 between the infected samples and an average correlation coefficient of R = 0.94 between the control samples, which indicated the high reproducibility and reliability of our experimental data (Figure S3).\\n\\n',\n",
       " 'Sx,y,z are calculated as follow: Sx=IPD(IPD2(Yl-Yr)(Yl+Yr)+4IPD(-XrYl2-XlYr2−(Xr+Xl)DSD2)+4(Xr2(Yl2+DSD2)-Xl2(Yr2+DSD2)))2(-2XrYl+2XlYr+IPD(Yl+Yr))2+\\xa08((IPD+Xl-Xr)2+(Yl-Yr)2)DSD2(1) Sy=2IPD(YlYr(−2XlYr+2XlYr+IPD(Yl+Yr))+(IPD+Xl−Xr)(Yl+Yr)DSD2)(-2XrYl+2XlYr+IPD(Yl+Yr))2+\\xa04((IPD+Xl-Xr)2+(Yl-Yr)2)DSD2(2) Sz=IPD(Yl+Yr)(-2XrYl+2XlYr+IPD(Yl+Yr))DSD+4IPD(IPD+Xl-Xr)DSD3(-2XrYl+2XlYr+IPD(Yl+Yr))2+4((IPD+Xl-Xr)2+(Y1-Yr)2)DSD2(3) From these components we calculated the saccade amplitude by Eq (4) azimuth (horizontal component of the eye movement), Eq (5) latitude (vertical component of the eye movement) and the focalized distance by Eq (6) vergence angles: Azimuth(α)=arctan(SxSz)(4) Latitude(α)=arctan(SySx2+Sz2)(5) Vergence(β)=arctg(DIP/2‖S‖)(6) Where ||S|| was calculated as: ‖S‖=Sx2+Sy2+Sz2(7) Only correct trials were analyzed. ',\n",
       " 'The most suitable model of DNA substitution for each locus and dataset was determined using MODELTEST 3.0 [38] according to the Akaike Information Criterion (AIC).',\n",
       " 'All possible CVD risk factors were included in the initial Cox proportional hazards model to explore the association between HOMA-IR and incident CVD events.',\n",
       " 'The subjects underwent prenatal follow-up, and the inclusion criteria included the following specifications: more than 18 years old, living in the urban area, not taking antidepressants or mood regulators and capable of understanding and completing the neuropsychiatric questionnaires in the postpartum period. ',\n",
       " 'path=journals/zma/2016-33/zma001016.a1en.pdf&mime=application/pdf&name=Attachment_13.08.15.pdf.\\n\\n',\n",
       " 'We checked whether there was an ordinal relationship between the genetic risk score and BMI category (i.e. thin, normal, or obese) using ordinal logistic regression with the clm function in the ordinal R package. ',\n",
       " '8.2 (SAS System for Windows, SAS Institute, Cary, NC, USA). ',\n",
       " 'In the last decade, several measures have been taken to reduce selective reporting in trials through the introduction of trial registration [31] and the Consolidated Standards of Reporting Trials (CONSORT) guidelines [32]. ',\n",
       " 'Transcriptional data were evaluated using Ingenuity Pathway Analysis (IPA, Systems Inc., USA) and prediction (increase of decrease) of biological activities occurring in the tissue was established. ',\n",
       " 'Here, we present an easily accessible software toolkit, called ENCORE, which can be used to compare conformational ensembles generated either from simulations alone or synergistically with experiments.\\n',\n",
       " 'All these analyses were performed using GraphPad Prism v7.0 (GraphPad Software, La Jolla, CA, USA).\\n',\n",
       " 'Numerical data used for all figures displaying MAGMA results are provided in S9 Table.\\n\\n',\n",
       " 'Data analyses were performed using SPSS 16.0.\\n\\n',\n",
       " 'In the 30-month questionnaire, the mother was asked how much time their child spent asleep at night (grouped into <10.5 or ≥10.5 hours), and in the 38-month questionnaire she was asked how much time they spent watching TV per week (grouped into ≤8 h or >8 h). ',\n",
       " 'Data were analyzed using GraphPad Prism version 5.0 for Mac OS X. All data are presented as mean ± SEM and evaluated for statistically significant differences (p <0.05) using a two-way ANOVA with Bonferroni’s post-hoc test, unless otherwise indicated.\\n\\n\\n',\n",
       " 'Coronary angiograms obtained before and after primary PCI were reviewed by two experienced interventional cardiologists and a report created on the Datacam software.',\n",
       " 'All statistical tests were conducted in Stata 11 (Stata Corp., College Station, TX [41]). ',\n",
       " 'Statistical analyses for incubation periods were performed using A Kruskal-Wallis test followed by Dunn’s multiple comparisons test using GraphPad Prism 7.0 for windows (GraphPad Software, La Jolla California, USA).\\n\\n',\n",
       " 'All statistical analyses were undertaken using R 3.1.0 (http://www.r-project.org).\\n\\n',\n",
       " 'The data that support the findings of this study are available from the corresponding author on reasonable request  .',\n",
       " 'The tree was visualized with the help of Treeviewer 1.6.6 (http://taxonomy. zoology.gla.ac.uk/rod/treeview.html) and rooted using ClustalW (http://www.bioinformatics.nl/tools/clustalw.html).',\n",
       " 'We present a general fragmentation methodology and accompanying tools called FragIt to help setup these calculations.\\n',\n",
       " 'For the analysis, we used the SPSS software for Windows, version 15.0 (SPSS Ltd.). ',\n",
       " 'All the processing, analyses, and machine learning were conducted using the MATLAB software package (version R2009a and R2015a, Mathworks Inc., Natick, MA, USA).\\n\\n',\n",
       " 'ENCORE is freely available and extendable, integrates with the established MDAnalysis software package, reads ensemble data in many common formats, and can work with large trajectory files.\\n',\n",
       " 'Individual-level covariates are based on administrative records, ICD-9 diagnosis and procedure codes, and Clinical Classifications Software (CCS) codes, developed by HCUP for use with ICD-9 codes. ',\n",
       " 'All datasets were available online, MtF as SPSS datafiles [30] and AF as data tables [31]. ',\n",
       " 'A software application for calculation of the Lewis score has been incorporated into the RAPID Reader platform for Pillcam SB.',\n",
       " 'Amplified DNA was analyzed for length variations on an ABI 3700 sequencer using GENEMAPPER 4.0 (Applied Biosystems).\\n\\n',\n",
       " \"Integrated conceptual information Φ (“big phi”) measures the difference between the conceptual structure C of the whole system S in state s0 and the conceptual structure CMIP of the partitioned system: The difference D between two conceptual structures is evaluated using an extended version of the earth-mover's distance (EMD), which quantifies the minimal cost of transforming the conceptual structure C of the whole to the conceptual structure CMIP of the partitioned set of elements. \",\n",
       " 'The covariance Σp is then the covariance of the set of the paused velocity profiles.',\n",
       " 'The AURIN platform has delivered such an integrated, data-rich research environment.',\n",
       " 'BioNetGen’s built-in ODE solver, CVODE from the SUNDIALS package [69], was then used to numerically integrate the ODEs, using default settings. ',\n",
       " 'The Gannet GABA analysis toolkit for Matlab was used to analyse GABA data [41].',\n",
       " 'Regardless of the initial cost, future publications should evaluate the hypothesis of a reduction of the global economic burden due to the application of autologous micro-fragmented adipose tissue in the treatment of DFU.\\n',\n",
       " 'The current experiment involved two new experimental stimuli: computer-generated images to be presented on a projection screen.',\n",
       " 'All statistical analyses were carried out with SPSS 19.0 software for Windows.',\n",
       " 'We keep 2 basis vectors with their corresponding activations to represent melodic content.',\n",
       " 'In the future, PID will allow users to take advantage of high-throughput protein-protein interaction data, either by allowing users to upload interaction sets to be added to the novel networks created by PID queries or by querying other data sources (such asPathway Commons, http://pathwaycommons.org) as needed to support a user query.',\n",
       " 'Models were fitted using the open source R programming language [34] and the H2O R interface version 3.8.2.2 [35] that optimizes all the analytical methods used for large datasets.',\n",
       " 'The resulting sequences were analyzed using Mothur v.1.33.3 [12].',\n",
       " 'To maximize reliability, variants with read depths < 10 and variants that failed quality control by GATK [84] were excluded. ',\n",
       " 'When the study was over participants were thanked, carefully debriefed and given their remuneration. ',\n",
       " 'Statistical analysis was done using IBM SPSS Statistics for Windows Version 22.0, Armonk, NY, IBM Corp. ',\n",
       " 'The normality tests, repeated measures and multivariate ANOVAs, Student t-test (two-tailed) and Mann-Withney analysis were done using SPSS statistic 19.0 (IBM). ',\n",
       " 'An installer is provided for Windows NT/XP/2003/Vista users.\\n',\n",
       " 'All data collected and analyzed within this study are available from the corresponding author on reasonable request.',\n",
       " 'PLINK v1.03 was used for the analysis [45].',\n",
       " 'All results reported are two-tailed and were performed with SPSS version 16.0.',\n",
       " 'Microarray data are available in the ArrayExpress database (www.ebi.ac.uk/arrayexpress) under the accession number E-MTAB-2249.',\n",
       " 'All of these sequences were compared by ClustalX (1.8; http://www.clustal.org) using global-multiple sequence alignment by the progressive method.',\n",
       " 'Analysis of Marker Gene Expression in Public Gene Expression Data: The selected marker genes used for qPCR were analyzed with the Condition Search tool “Perturbations”in Genevestigator https://genevestigator.com/gv/doc/intro_plant.jsp [31].',\n",
       " 'Designed by and for biologists, CellProfiler equips researchers with powerful computational tools via a well-documented user interface, empowering biologists in all fields to create quantitative, reproducible image analysis workflows.\\n',\n",
       " 'All statistical procedures were completed using SPSS 16.0 (SPSS, Chicago, IL, USA). ',\n",
       " 'Descriptive statistics were performed using Graphpad Prism 6.',\n",
       " 'The lenses were worn in the right eye, while the left eye was fully corrected for distance with a trial frame and served as the control eye. ',\n",
       " 'The response rates were about 76% (WOBASZ) and 45.5% (WOBASZ II).',\n",
       " 'However, risk of MHV transmission by sperm or oocytes (IVF) or by embryo transfer seems to be low, though thorough washing of gametes and embryos is required [211], [340], [341], [342].\\n',\n",
       " 'All mice were housed in a temperature-controlled facility.',\n",
       " 'Information from completed abstraction forms were entered into an Access database and imported into SAS for analysis.\\n\\n',\n",
       " 'Samples were analyzed by a FACScan cytofluorimeter using the CellQuest software; fluorescence intensity was expressed in arbitrary units on logarithmic scale.\\n\\n',\n",
       " \"Connectivity evolves indirectly during adaption to the environment as outlined below, following a genetic algorithm that selects, mutates, and updates the animat's genome at each new generation. \",\n",
       " 'We followed the advice on the STAC website (cbil.upenn.edu/STAC/) and divided each chromosome arm into stretches of 1 Mb.',\n",
       " 'We analyzed data using JMP® v.8.0 (SAS Institute Inc., Cary, NC), comparing males and females for all variables using a t-test and considered results significant if α≤0.05 [47]. ',\n",
       " 'SPSS (version 24) was also used to analyse response times with repeated measures analysis of variance (ANOVA).',\n",
       " 'In order to further identify a set of likely independent signals we performed conditional analysis of the lead SNPs in SNPTEST to take into account long-range LD.',\n",
       " 'Tests for linkage disequilibrium were checked by using the GenePop v.4.0 [62].',\n",
       " 'Descriptive statistics of survey data were performed in STATA (Version 13; STATA Corp, College Station, TX, USA). ',\n",
       " 'Podbat requires Java version 1.6 or higher.\\n',\n",
       " 'The newey-west regression and robust regression in the Stata were used to correct the autocorrelation and heteroskedasticity.',\n",
       " 'Note that the elements are not limited to classic logic gates, such as ANDs, ORs, or XORs, but can potentially specify any deterministic logic function over their inputs. ',\n",
       " 'The Statistical Package for the Social Sciences (SPSS Inc., Chicago, IL) was used for statistical calculations. ',\n",
       " 'We used STATA 13.0 (College Station, TX) for all analyses.\\n\\n',\n",
       " 'Since 2012, MTB has been downloaded by 1104 researchers.',\n",
       " 'The team involved in the computational analysis was not informed about any clinical information until the end of the study, when computational and clinical data were matched.',\n",
       " 'The study was approved by the Health Research Authority East Midlands–Nottingham 1 Research Ethics Committee (13/EM/0255), and parents/guardians gave written informed consent (children gave assent as appropriate). ',\n",
       " 'The selected maps were then clustered using K-Means, as implemented in Scikit-learn 0.17 [34]. ',\n",
       " 'The interaction term was convolved using the AFNI waver command with a basic gamma function.',\n",
       " 'Here we propose a new approach to transmission inference called SCOTTI (Structured COalescent Transmission Tree Inference).\\n',\n",
       " 'The first 2 min were excluded from this analysis because during the first 2 min, system calibration and establishment of the units’ thresholds across the recording channels at the beginning of each recording sessions were performed. ',\n",
       " \"Results were then fitted with cumulative Gaussian psychometric functions using custom Matlab code, to give a value for each subject's point of subjective equality (PSE) for the speed of smooth radial flow that matched the perceived speed of the oscillating flow. \",\n",
       " 'The new DIANA-microT web server aims to facilitate users, not having access to extensive computational infrastructures and support, to perform ready-to-use sophisticated pipelines.',\n",
       " 'Availability: FunciSNP is available from Bioconductor (bioconductor.org).\\n',\n",
       " 'We describe our scalable framework gpu-lasso, which can be particularly useful for fitting sparse models in high-dimensional settings. ',\n",
       " 'To address these issues we developed the interactive DREM (iDREM) tool that provides support for more data types and greatly improves the visualization allowing users to interactively query the reconstructed network.\\n',\n",
       " 'Statistical analysis was performed using SPSS version 19.0 (SPSS Inc., Illinois, USA) and the significance level was set at P<0.05. ',\n",
       " 'The discrimination subtest also uses a 3-AFC procedure: triplets of pens are presented in randomized order, with two containing the same and one a different odorant. ',\n",
       " 'For experiments included in groups',\n",
       " 'Analyses were performed using the SPSS Windows Version 20 [33].',\n",
       " 'Figure 1E\\n',\n",
       " 'Analyses were conducted using Stata v11.',\n",
       " 'All loci were collected as an EXCEL file, and then converted into the computable files using convert [61]. ',\n",
       " 'The completed questionnaires were coded and entered into EPI info version 3.5.3 statistical software and then exported to SPSS windows version 20 program for analysis.',\n",
       " 'All statistical analyses were conducted using Statistical Package for Social Sciences (SPSS 20.0).\\n\\n',\n",
       " 'SPSS version 16 for Mac OS 10.5 was used to run the linear mixed effect models, significance levels were set at 0.05, and two-tailed probability values are quoted.\\n\\n',\n",
       " \"We have deliberately chosen to study routines which span what Goffman calls 'front-stage' work (e.g. carried out by clinicians in their consulting rooms) and the 'back-stage' activities which support and augment this work [39].\",\n",
       " 'We carried out imputation to HapMap release 22 using Mach 1.0, Markov Chain Haplotyping [16].\\n\\n',\n",
       " 'We released 5 genotypes, 4 exomes, 1 metagenome from the Corpas family via a blog and figshare under a public domain license, inviting scientists to join the crowdsourcing efforts to analyse the genomes in return for coauthorship or acknowldgement in derived papers.',\n",
       " 'Data analyses were performed in SPSS (SPSS Statistics for Windows, Version 20.0. Armonk, NY: IBM Corp) and confirmed in R (Version 3.0.1, The R Foundation for Statistical Computing, Vienna, Austria). ',\n",
       " 'The level of IL-1α, IL-1β, IL-6, IL-8, TNF-α and VEGF was calculated by the Bioplex Manager software based on the recorded 7-point calibration curve. ',\n",
       " \"If the matching fails, TFPM can be simplified based on the object’s positional relationship between adjacent frames (see 'Motion association' section).\\n\\n\",\n",
       " 'In contrast to the original tool, ggsashimi internally generates an R script which uses the ggplot2 library [5] for the graphical rendering.\\n',\n",
       " 'adoption of the universal tracking and of the antibiotic prophylaxis resulted in an\\n',\n",
       " 'All analyses were performed using Statview software (SAS, Cary, NC), with significance accepted at P≤0.05.\\n\\n',\n",
       " 'SCR data were acquired continuously using flat-surface electrodes placed on the non-dominant hand and stored for off-line analysis using a second computer running Chart v4.2 software.',\n",
       " 'Data were analyzed using version 17.0 version of SPSS software.',\n",
       " 'To demonstrate the utility and use of MicroSyn, we presented a case study of the evolution of miR167 gene family in Arabidopsis thaliana (Arabidopsis), Populus trichocarpa (Populus), Vitis vinifera (grape) and Oryza sativa (rice).\\n',\n",
       " 'Risk ratios (RR) and 95% Confidence Intervals (CI) were calculated by using Review Manager 5.1.\\n\\n',\n",
       " 'The exon arrangement of the S. crispa β-glucan synthase gene was analyzed by FGENESH software (http://linux1.softberry.com/) [21] using Coprinopsis cinerea genome as a reference.',\n",
       " 'Here, we present a new cell image analysis software platform called \"cellXpress\" for high-throughput phenotypic profiling.\\n',\n",
       " 'The frozen brain tissue regions were weighted and transferred to an Eppendorf tube.',\n",
       " 'ECM3 was identified using the Large Average Submatricies (LAS) biclustering method [25]; compactness and separation of cluster partitions were evaluated using the Dunn index and silhouette width [26]; connectedness of clusters was quantified using the connectivity measure [26], as previously described [4].\\n',\n",
       " 'Because there are insufficient data from a single country in Africa for every input in RabiesEcon, we used data from a number of African countries, primarily Chad, Malawi and Tanzania (Table 1). ',\n",
       " 'The high resolution scans made available (Harvard Dataverse 10.7910/DVN/29628, [2]) were produced using 3600 radiographs with 1.2 µm effective pixel size.',\n",
       " 'Reads’ 5′-end counts were normalized to local background in rolling window of 10 nucleotides, values for U residues were excluded to calculate the median. ',\n",
       " 'Such a normalization step is not implemented in GRAB [14] suggesting that GRAB might be sensitive to ascertainment bias and general population diversity.',\n",
       " 'These procedures were performed using the SPSS 11.5 software package [40].\\n\\n',\n",
       " '(The total altmetric score shows in the middle of the donut graphic.)',\n",
       " 'The response onset to the prime stimulus was defined as the time from which the velocity differences between RR (LL) and LR (RL) continuously exceeded a threshold value of 30 mm/s for at least 40 ms in a window between 150 and 300 ms after the prime onset.',\n",
       " 'SAS (version 8) was used in descriptive analysis of the sample demographics.',\n",
       " \"Location of activated clusters was verified using a printed atlas [27] and labelled using Tailarach's nomenclature [23] by means of the Tailarach Daemon (Research Imaging Center, University of Texas Health Center, San Antonio, TX, USA). \",\n",
       " 'Stimuli were presented with E-prime software (Psychology Software Tools, Pittsburgh, PA) on a personal computer, back-projected onto a screen using a liquid crystal display projector and viewed by participants through a mirror mounted on the MRI head coil. ',\n",
       " 'In all cases, 95% confidence intervals were constructed, and an alpha = 0.05 was considered significant.\\n\\n',\n",
       " 'OHM (OligoHeatMap) is an online tool able to provide estimates of T for a set of oligomers and a set of aligned sequences, not only as text files of complete results but also in a graphical way: T values are translated into colors and displayed as a heat map image, either stand alone or to be used by softwares such as TreeDyn to be included in a phylogenetic tree.\\n',\n",
       " 'We have developed a tool, called MicroSyn, to semi-automatically detect microsynteny within an individual gene family.\\n',\n",
       " 'Hierarchical clustering was done using Partek’s default clustering method.',\n",
       " 'SPSS statistics for windows (version 17.0, Chicago, USA) was used for analysis. ',\n",
       " 'A computer software, AccuTyping, was developed based on the algorithms described above.\\n',\n",
       " 'The bulk of the literature typically concerns pairwise comparisons between spatially defined populations based on the (di)similarity of their respective surname compositions [5], [6], [14], [15]. ',\n",
       " 'We developed a software package, V-Phaser 2, for inferring intrahost diversity within viral populations.\\n',\n",
       " 'Graphpad prism 7.01 software (Graphad Software) was used for statistical analysis.',\n",
       " 'MATLAB code for computing the threshold likelihood has been posted on MATLAB Central and complete MATLAB code for all analyses and simulations can be found in supplementary material, Supplementary Material online.',\n",
       " 'Immunoratio [16], for example, is an open-source software based on ImageJ [17], another open-source software available as either a stand-alone or Web application. ',\n",
       " '(ii) ten pure farmed families; and',\n",
       " 'We do this by Maximum Likelihood, using Matlab (the program is available in https://www.york.ac.uk/economics/exec/research/zhouandhey1/).',\n",
       " 'Although α accounts for multiple factors that simultaneously contribute to quorum-sensing dynamics, to aid intuition, one may hold all parameters other than φ constant and think of α as the critical quorum-sensing autoinducer concentration. ',\n",
       " 'The Skeeter Buster model simulates the biological development of four life stages of Ae.',\n",
       " 'ProDySummary: We developed a Python package, ProDy, for structure-based analysis of protein dynamics.\\n',\n",
       " 'We presented a novel algorithm called DriverNet for integrative analysis of genomic and transcriptomic data derived from population-level studies of tumors.\\n',\n",
       " 'Keyword searches can be made using the EBI-Search engine (16) through both a web form (http://www. ebi.ac.uk/ebisearch) and the corresponding SOAP web services.',\n",
       " 'In our context the online communication software Skype (http:// www.skype.com) and the word processing application Google Docs (http://www.google.com/google-d-s/intl/de/ tour1.html) were used.',\n",
       " 'The GMM approach utilises instruments for the estimation of Eq (5).',\n",
       " 'The likelihood-based, COLONY 2 program [51] was used to analyze genetic relationships between the attending males, feeding females and the offspring in the nests. ',\n",
       " 'These were compared to predicted responses (Fig 1D) obtained by convolution of the STRF with the validation stimulus (MATLAB conv2 function; [16,23,40,62]). ',\n",
       " 'Genotype calling was performed centrally for all batches at the WSI using GenCall.',\n",
       " 'Analyses were conducted using STATA Version 13 for Windows (College Station, TX, USA).\\n\\n',\n",
       " 'BRASS has been developed by Dr Brown PE and co-workers and is available from http://millar.bio.ed.ac.uk/Downloads.html.',\n",
       " 'The software and documentation can be downloaded at: http://ski.clps.brown.edu/hddm_docs/\\n',\n",
       " 'We compared runtime behavior of our program across multiple configurations and to glmnet (Friedman et al., 2010). ',\n",
       " 'Relative metabolite concentrations were determined using Chenomx software by manual fitting of the proton resonance lines for the compounds available in the database.',\n",
       " '[27] STATA 13.0 was used for analysis in 2015, using the svy suite, given the sample design.\\n\\n',\n",
       " 'Chaste is developed using industrially-derived techniques, in particular test-driven development, to ensure code quality, re-use and reliability.\\n',\n",
       " 'Prior to commencement, we established the reliability of the clinical interpretation of digitally acquired images by performing a pilot study assessing the ability of a clinical nurse specialist (CNS) in wound care to assess a digital image. ',\n",
       " 'For accession IDs of amino acid sequences used in this analysis, see supplementarySea lamprey P. marinus ENC-A was predicted in the AUGUSTUS web server (http://bioinf.uni-greifswald.de/ webaugustus/prediction/create, last accessed July 24, 2013) with its species-specific parameters on the supercontig22564 in the version 3 assembly of the genome sequencing project (PMAR3.0).',\n",
       " 'Experimental inoculation studies show that several MNV strains are able to persist in various tissues (small intestine, caecum, mesenteric lymph node, spleen) of immunocompetent (C57BL/6J, Hsd:ICR, Jcl:ICR) and immunodeficient (CB17-Prkdc) mice with viral shedding in faeces for the duration of at least 35-60\\xa0days [367], [368], [369]. ',\n",
       " 'ISCs were calculated for data mapped into the common model connectome, for movie data aligned anatomically, for HCP resting state data aligned with MSM-All, and for filter control movie and HCP data. ',\n",
       " 'NQuery Advisor by StatsSols version 3.0 was used to calculate sample size.',\n",
       " '(i) Visual inspection with dynamic varying of thresholds using a software such as Mricron or FSLView.',\n",
       " 'The specific components coded for each case are identified in Table 2.\\n\\n',\n",
       " 'Main effects in the model included area (the four MPA areas surveyed), site (MPA or REF) and sample year as a continuous covariate. ',\n",
       " 'In short, there are three advantages for illustrating our results with Pong: ',\n",
       " 'The religious belief scale was administered in online mass testing surveys with a number of different scales.',\n",
       " 'Similar to the MGH-HPS, the Skin Picking Scale-Revised (SPS) was also modelled after the Y-BOCS [10]. ',\n",
       " 'The simulations were run in Excel from Microsoft Office.',\n",
       " 'Data were analyzed in Microsoft Excel 10.0 (Microsoft, Redmond, WA) and STATA/IC 12.1 (Stata Corp, College Station, TX).\\n\\n',\n",
       " 'To avoid collinearity issues, the covariates of model 3 were tailored to each frailty score, excluding covariates that were an underlying variable of the score or a highly correlated variable. ',\n",
       " 'Variant calling: The GATK haplotype caller (version 3.2) was chosen to perform multiple sample variant calling and genotyping with default parameters.',\n",
       " 'Data were expressed as means ± S.E. All statistical calculations were made using TIBCO Statistica software v.13.3 (TIBCO Software Inc.).\\n\\n',\n",
       " 'All statistical tests were analysed using IBM SPSS version 22.',\n",
       " 'Each model process was programmed in Matlab (Mathworks) and fitted to data for each observer and condition of change probability.',\n",
       " 'First, two sets of variables were selected and then the Pearson’s correlation test performed with the software ENMtools v 1.3 [31] was used to ensure the lack of multicolinearity among the selected predictors [32]. ',\n",
       " 'We modeled group size for wildebeest and zebra as a function of habitat type, season and lion UD using generalized linear models in SPSS 15.0. ',\n",
       " 'Stained cells were analyzed on a FC500 flow cytometer (Beckman Coulter), with data analyzed with FlowJo software (Tree Star, Ashland, OR, USA).\\n\\n',\n",
       " 'Although Immunoratio obtained a hazard ratio of 2.2 in survival analysis, like all algorithms using standard IHC preparations, it is disadvantaged by the observation of lower Ki67 LI values purportedly because of normal stromal and inflammatory cells that artificially increase the denominator in LI assessments [18]. ',\n",
       " 'In addition to providing functional clues on neuroblastoma pathogenicity, the secretome protein fraction may also contain potential biomarkers which could be exploited as indicators of the disease condition. ',\n",
       " 'All data were analysed using SPSS statistics version 17.0.0 released Aug 23.',\n",
       " 'We further tested the equality of Harrell’s C-index concordance areas of the pair-wise comparable discriminatory ability of the best-fit anthropometric measure with each of the other anthropometric measures (six pairs) within race-gender groups, by examining the Harrell’s C-index and its p value [17] with Bonferroni correction for multiple testing [18] Statistical analyses were conducted using Stata MP, Version 14.0 (StataCorp, College Station, Texas, U.S.). ',\n",
       " 'In reality, as the preliminary data fragments in Table [1] show, the ostensive, performative and proxy routines for scanning and coding incoming letters are social constructions which are differently perceived by different organisational members. ',\n",
       " 'The work reported in this article was mentioned in the following extracts:\\n',\n",
       " 'The experiment was programmed in Python and interfaced with the eye-tracker.',\n",
       " 'All further statistical analyses (t-tests for dependent samples) were calculated using the software package SPSS 11.5 (SPSS Inc., Chicago, USA). ',\n",
       " 'There was no record of description, of either radiographic or clinical aspects of this disease, until Röske (1930) [3] described an individual case similar to ICH, which he could not classify.\\n',\n",
       " 'Cases of the GHEA cohort [23] were submitted to LAS biclustering to identify ECM3 tumors.',\n",
       " 'The geographic distance between the entrepreneurs and the donors was obtained for the cities where they were located, and we used Google Maps APIs (http://code.google.com/apis/maps/) to identify the coordinates of each location.',\n",
       " 'The R GeneOverlap package (https://bioconductor.org/packages/release/bioc/html/GeneOverlap.html) was used to compute the p-value.\\n\\n',\n",
       " 'Finally, the data were represented as a filled contour plot using graphing software (SigmaPlot 9.0 for Windows; Systat Software, Inc., Richmond, CA) that constructs pseudocolored isodensity maps in a scale of 45 different steps (each of 125) ranging from 0 to 5,625 cells/mm. ',\n",
       " 'The analyses were conducted using the SPSS statistical software program (PASW 18.0).\\n\\n',\n",
       " 'The MuPeXI webserver [43] was then used to extract 8 to 11 length peptides around missense mutations, indels and frameshift mutations from the somatic VCF files and all mutant peptides with a binding prediction to MHC below 2% (weak binders) from NetMHCpan [44] were retained as neoantigens.',\n",
       " 'More details:Software name: CBFA plugin for Optflux Project home page: http://www.optflux.org/ Methods details and application tutorial: http://www.optflux.org/cbfa Operating system(s): Platform independent Programming languages: Java Other requirements: Java JRE 1.7.x (for Mac OS users the installation of JDK 1.7 is recommended), GLPK License: GNU-GPL, version 3\"\\n',\n",
       " 'All statistical analyses were conducted using SPSS Statistics 22 (Dynelytics, Zurich, Switzerland). ',\n",
       " 'In addition, new repetitive elements were generated from the Haruna Nijo genome assembly by RepeatModeler (http://www.repeatmasker.org/RepeatModeler.html, 7 November 2015, date accessed).',\n",
       " 'All statistical analyses were conducted using the SPSS version 17.0 (IBM Inc., Armonk, NY, USA). ',\n",
       " 'Thus, we evaluated the effects of different doses of perifosine (0.5, 1.0 or 2.5 μM) alone and in combination with sorafenib (10 and 20 μM) in T24 BC cells. ',\n",
       " 'To fill this gap, we developed a web-based application, QuIN, which enables: ',\n",
       " 'We used Aninhado [3] to analyse the existence of nestedness patterns (nestedness as NODF estimator), generating 1000 replicates for each saproxylic sub-network with a CE null model. ',\n",
       " 'Multiple therapies [5] for no reflow have been tested in animals and to a lesser degree in humans. \\n',\n",
       " 'We carried out meta-analyses of the results from the two cohorts using two methods in METAL (www.sph.umich.edu/csg/abecasis/metal).',\n",
       " 'After, we used the cost-distance grids as inputs for the Corridor function in Spatial Analyst for all proximate pairs of PJCUs, resulting in least-cost corridors among each pair. ',\n",
       " 'All analyses were performed with statistical software STATA version 9.0 (StataCorp, College Station, TX).\\n\\n',\n",
       " 'All analyses were undertaken using the IBM SPSS 22 software.\\n\\n',\n",
       " 'The datasets used and/or analyzed during the current study are available from the corresponding author on reasonable request.',\n",
       " 'The Institut Pasteur, RPBS and LIPM Mobyle servers have been publicly available since January 2008, October 2008 and January 2009, respectively.',\n",
       " 'All statistical tests, carried out in R Studio Version 0.98.1102 and SPSS (IBM SPSS Statistics for Macintosh, version 22.0), were two-tailed and the statistical significance level was set at 0.05.\\n\\n',\n",
       " 'From these values one can compute V (which is directly proportional to A) and N. Then V, A, and N are recalculated after changing the threshold value in a given range around the initial T. The goal is to observe the magnitude of variation of both A and N and consequently robustness of N and V. The whole simulation is performed using macro scripting facility of ImageJ. ',\n",
       " 'You can get the most recent virtual appliance by browsing to: https://sourceforge.net/projects/physicell/files/PhysiCell/.\\n',\n",
       " 'Optimized BNC and MN masks were developed by applying and combining several new and existing mask functions in IDEAS (version 6.2).',\n",
       " 'Furthermore, histologic features of small and larger retinal tumors are identical to those of nerve root and cerebellar hemangioblastomas. ',\n",
       " 'The data were analyzed using statistical software STATA version 13 and statistical package for social science (SPSS) version 22.',\n",
       " 'Stimuli were elaborated using the image processing toolbox on MATLAB (Mathworks Inc., Sherborn, MA, USA). ',\n",
       " 'The alignment of raw reads was conducted by Bowtie2 ( [53]) in end-to-end mode.',\n",
       " 'After each demonstration was presented, the participants wrote down their responses; when all participants had finished recording their responses to a demonstration, the next demonstration was presented.\\n\\n',\n",
       " 'Definition of masks and searchlights for movie data: We derived a gray matter mask by segmenting the MNI_avg152T1 brain provided in AFNI and removing any voxel that was outside the cortical surface by more than twice the thickness of the gray matter at each surface node.',\n",
       " 'The search was conducted using PAUP* v.4.0b10 [45] with 100 random stepwise additions and TBR branch swapping.',\n",
       " 'All the analyses were performed using the Stata 13 statistical program (Stata Corporation, College Station, TX, USA).\\n\\n',\n",
       " 'Statistical analyses were performed with IBM SPSS 23.0 Statistics for Windows (SPSS Inc., Chicago, IL, USA) or GraphPad Software. ',\n",
       " 'For each participant, mean amplitudes of the components of interest (see above) were exported and read into SPSS (version 21, IBM). ',\n",
       " 'The AR system \"OnlineTED\" can be accessed via the openaccess website http://www.OnlineTED.com (German version: http://www.OnlineTED.de).',\n",
       " 'Transcript counts were produced by Sailfish version 0.6.3.',\n",
       " 'Patients also answered clinical questions regarding their age of first opioid use and any secondary drugs of abuse.',\n",
       " 'Subjects received cash for their performance after the experimental sessions, but not the practice sessions. ',\n",
       " 'Immunoratio has shown correlation with a visual LI assessment using a median Ki67 LI of 20% on a standard Ki67 IHC stain using the chromogen 3,3′-diaminobenzidine (DAB) and hematoxylin as the counterstain. ',\n",
       " 'The resulting mean values for the activated regions were entered in an ANOVA (SPSS 11.5) comparison of the different conditions and regions.\\n\\n',\n",
       " 'All analyses were conducted using Stata 12.1 [44].',\n",
       " 'Altogether, these data confirm that our culture conditions are suitable to sustain the production of ASC from both SAT and VAT and to preserve their intrinsic functional differences.\\n\\n',\n",
       " 'fMRI data were analyzed using SPM8 (Wellcome Trust Centre for Neuroimaging, http://www.fil.ion.ucl.ac.uk/spm). ',\n",
       " 'For data analysis we used standard FSL tools v5.0.',\n",
       " 'The gene is translated as a full length gp160 precursor protein and undergoes secondary structure formation and glycosylation in the endoplasmic reticulum [10].',\n",
       " 'All statistical tests were done with SPSS (v. 22).\\n\\n',\n",
       " 'And a powerful gene finding program, BGF (Beijing Gene Finder), is proposed by Beijing Genomics Institute (http://bgf.genomics.org.cn/).',\n",
       " 'We use Telescope to investigate the expression of HERVs in cell types from the ENCODE consortium.\\n',\n",
       " 'Analyses were performed with the statistical software SPSS v.21 and in all cases was established as statistically significant a p-value < 0.05.',\n",
       " 'We used Support Vector Machine (SVM) with linear kernel (using Matlab functions “svmtrain” and “svmclassify”) as multivoxel pattern classifier for discriminating between social vs. individual action intention.',\n",
       " 'A phylogenetic tree was reconstructed with maximum likelihood using the PhyML program (www.atgc-montpellier.fr/phyml/), and robustness was assessed by the bootstrap method (100 pseudoreplicates). ',\n",
       " '[15].',\n",
       " 'For phylogenetic reconstruction, sequences were aligned with Muscle (www.ebi.ac.uk/Tools/msa/muscle/). ',\n",
       " 'This classification module was incorporated into our karyotyping software tool, called MetaSel, which was written from scratch using C# on Microsoft Windows 7 operating system.\\n',\n",
       " 'The CGBayesNets package is available from the authors and via anonymous download from www.cgbayesnets.com.\\n',\n",
       " 'The SEER stage was based on the time of diagnosis.\\n\\n',\n",
       " 'For quantitative analysis, fluorescent areas were measured using the ImageJ 1.44 software (available at http://rsbweb.nih.gov/ij/).',\n",
       " 'This paper describes the optimization of the IDEAS analysis template for identifying and scoring BNCs and MN in the IS X -based version of the CBMN assay.',\n",
       " 'However, one study showed that while 50% of the study population was infected with H. pylorus, the proportion of patients infected with H. pylorus who went on to develop gastric cancer was <5% [8]. ',\n",
       " 'D-prime, which measures the distance between the signal and the noise means in standard deviation units (40), was calculated using matlab function dprime_simple (https://it.mathworks.com/matlabcentral/fileex change/47711-dprime-simple-m; date last accessed April 1, 2017).',\n",
       " 'At age 5;8, the proband underwent an occupational therapy evaluation that revealed severe fine motor deficits, especially in grasping and eye-hand coordination skills, qualifying him for services in this area, whereas his gross motor development was found to be within normal limits. ',\n",
       " 'The word ‘hello’ was extracted from each recording, and normalised for power (RMS) and loudness via Matlab (the Mathworks). ',\n",
       " 'After m principal components t1, t2, ⋯, tm are finally extracted from X, PLS first performs a regression of yk on t1, t2, ⋯, tm and converts it in the regression equation of yk on x1, x2, ⋯, xp. ',\n",
       " 'A subroutine developed with the IPP image analysis program counted the number of dots (RT97 RGCs) and their topological distribution within each analyzed retina [26,74]. ',\n",
       " 'The very first strip chart (A) illustrates the V results, whereas the last one (E) is the Immunoratio results. ',\n",
       " 'Reads were quality fi ltered, and the adapters were removed using libngs (https://github.com/ sylvainforet/libngs) using a minimum quality of 25 bp and a minimum read size of 150 bp.',\n",
       " 'Convenient integration between Matlab and Google Cloud was achieved through using a software called Techila Technologies.',\n",
       " 'Statistical analysis and figures were completed in R version 2.15.3 [10].',\n",
       " 'Variance components, error estimates, and reliability coefficients were derived using EXCEL macros created by the corresponding author (see Tables S1 and S2 for a detailed description of the G theory calculations).\\n\\n',\n",
       " 'All statistical analyses were carried out with the use of Stata statistical software (version 9.2; Stata Corporation).',\n",
       " 'In this study, we describe the development ofpssRNAMiner, a web-based server which identifies ta-siRNA clusters as well as their potential phaseinitiators.',\n",
       " 'The best-fitting nucleotide substitution model for each gene (16S:GTR+I+G, Cytb: GTR+G, D-loop: HKY+I+G) was selected based on the Akaike Information Criterion as implemented in Modeltest v.3.7 [50] using default parameters. ',\n",
       " 'A participant was considered as physically active if he/she reported practicing at least 2 hours of leisure-time physical activity per week.',\n",
       " 'Measurement of cortical thickness and hippocampus volume: After sections were Nissl-stained, images were captured with a video camera 3CCD (DXC-9300; Sony) coupled to an image analysis system (Visilog 6.3; Noesis).',\n",
       " 'However, once the dendrite formation occurred during cycling, Li dendrites grew up out of control ( and ). ',\n",
       " 'All figures were drawn using Python 2.7.5 and Matplotlib 1.5.0 (https://matplotlib.org/).',\n",
       " 'ImageJ (Fiji) was used for analysis of positive cells.\\n\\n',\n",
       " 'Images were analyzed with the ImageJ analysis program (National Institute of Health).\\n\\n',\n",
       " 'The software is written as a native plugin in Objective-C and is integrated within the mixed C/C++/Objective-C environs of OsiriX.',\n",
       " 'We collected the fecal samples from the Henan Taihang Macaque Nature Reserve, and the Zhangjiajie National Forest Park. ',\n",
       " 'The unmapped reads were subsequently mapped to Reference genome hg19 from the UCSC genome browser (www.genome.ucsc.edu/).',\n",
       " 'ij\\n',\n",
       " 'To extract noun phrases to use as search terms for each of the papers on PubMed Central, we run a software program written in Python based NLTK (http://www.nltk.org/) to automatically extract nouns, adjectives, and noun phrases from each sentence in the paper. ',\n",
       " 'For model level inference, we calculated exceedance probabilities (i.e., the probability that model A is better at explaining the observed data than model B) using random effects Bayesian model selection [17] as implemented in SPM12 (http://www.fil.ion.ucl.ac.uk/spm/software/spm12/). ',\n",
       " 'They were implementation research projects with intervention arm.',\n",
       " 'Here, we introduce PyPhi, a Python software package that implements this framework for causal analysis and unfolds the full cause-effect structure of discrete dynamical systems of binary elements.\\n',\n",
       " 'Thus, Mars 520 had many essential features of an isolated and confined environment (ICE) that had the fidelity necessary to study behavioral and psychological reactions to prolonged space flight. ',\n",
       " 'Among all, we selected ODEs since they represent the dynamical states of small networks more precisely along the time. ',\n",
       " 'This study was based on the analysis of existing datasets in the Demographic and Health Survey Program repository that are freely available online with all identifier information removed (https://dhsprogram.com/data/available-datasets.cfm).',\n",
       " 'During each trial, startle probes were presented 3 s after picture onset during the first (pre-instruction) and the second (post-instruction) picture-viewing phases. ',\n",
       " 'To assess whether relationships based on perceptions differed from those based on objective data, we ran a survey using Amazon Mechanical Turk (https://www.mturk.com) to explore attitudes towards blood donations across 78 high-and low-income countries.',\n",
       " 'The eye-tracking data were processed with custom software written in MATLAB (Mathworks, Natick, MA, USA).\\n\\n',\n",
       " 'Herein we present PoreWalker, a method to provide a detailed description of the three dimensional geometry of a channel (or pore) through a transmembrane protein, given the coordinates of the protein structure.\\n',\n",
       " 'Data were analyzed with the Statistical Package for the Social Sciences (SPSS) for Windows (Version19.0).',\n",
       " 'The crystal structure of JAK2 in complex with a potent quinoxaline inhibitor, PDB:3KRR [52], was downloaded in MOE. ',\n",
       " 'We used a similar formula to calculate the cost per YLG, assuming that the average age of dog-rabies related death is 10 years of age [28], and that life expectancy at age 10 is approximately 53 years [27] (Table 1) (Additional details in S2 Appendix, Note #2).\\n\\n',\n",
       " 'All analyses were performed with SAS software 8.2 (USA).',\n",
       " 'Non-redundant sequences were then aligned to genomic RNA and mRNA sequences of hg19 using Bowtie 2 (www. bowtie-bio.sf.net) (35) and matched sequences were retained and aligned to common and abundant non-coding RNAs (tRNAs, rRNAs, snoRNAs).',\n",
       " 'h) between large and small fish on the three substrata was compared using Survival Analysis (Statistica 8.0).',\n",
       " 'The pre-processed data files were imported into Agilent Mass Profiler Professional software (version 12.1) for further statistical analysis.',\n",
       " 'Both independent variables were between subjects.',\n",
       " 'The top blastn hits against the GenBank nucleotide database were determined for representative sequences of each of these 11 OTUs.',\n",
       " 'Final file formatting, de-multiplexing, and fastq generation were carried out using CASAVA v 1.8.2.',\n",
       " 'For outcomes collected at the 2-, 4-, and 6-mo visits, we explored whether the effect of the trial garments on the outcome changed over the study period by including an interaction term between treatment group and time point in the model. ',\n",
       " 'We used Stella [16] integrated debugger to detect the memory locations containing information about these events.',\n",
       " 'Based on these sig-mers, we developed a method, RNA-Skim, which is much faster than Sailfish and also maintains the same level of accuracy in the results.\\n',\n",
       " 'The following search strategy was used for MEDLINE: cumulative adj10 meta?',\n",
       " 'All relevant data are included in our paper.',\n",
       " 'The signal intensity was evaluated using Quantity One software (Bio-Rad).\\n\\n',\n",
       " 'Additional tests were conducted using SPSS 19.0, and all tests were two-tailed. ',\n",
       " 'All statistical analyses of this study were performed using SAS version 9.3 and STATA IC/SE 14.\\n\\n',\n",
       " 'The granule cell area was traced using an image analysis system (Visilog 6.3; Noesis).',\n",
       " '3. We used a custom-made Matlab (Mathworks) script to cross-correlate the propagated recordings with the original sequences.',\n",
       " 'As more than one population has been interpreted in the mudwasp data, we have analysed the grains using a Finite Mixture Model (FMM) [49]. ',\n",
       " 'This baseline of ~0.2 Gy agrees with the lower component of the FMM for all the nests ranging from 0.15-0.34 Gy.',\n",
       " 'Immunocytofluorescence studies\\n',\n",
       " 'While users can specify the feature to be used, MaxEnt automatically selects individual features for each predictor that contribute most to model fit using regularization coefficient β [23,26]. ',\n",
       " 'The tubing fed through a hole in the table (immediately under the bowl) and connected to the pump and then to a reservoir of soup via a hole in the screen.',\n",
       " 'That is, if all six items were included in the analysis a similar result, but with a somewhat smaller effect size, was still evident when comparing recollection of choice blind items with non-manipulated items, χ2 (2, n = 196) = 92.4, p < .001, Cramer’s V = 0.69, p < .001.\\n\\n',\n",
       " 'To create the actual movie files, in house made scripts were used, created with Matlab software (MathWorks, Massachusetts, U.S.A.). ',\n",
       " 'Data were analysed using Stata V13 (StataCorp, Stata Statistical Software: Release 13, College Station, Texas, USA). ',\n",
       " 'The raw data from the analysis were analyzed using Coffalyser.NET (beta version, MRC-Holland, Amsterdam, the Netherlands).\\n\\n',\n",
       " 'All statistical analyses in the present study were performed with MATLAB [23].\\n\\n',\n",
       " 'License: GNU GPL2\"\\n',\n",
       " 'These were all screened for relevant articles through 4 December 2012.',\n",
       " 'Culture of Cancer Cell Lines: The authenticated (RADIL, Columbia, MO), human pancreatic cancer cell lines BXPC-3 (witout ras mutations) and Panc-1 (expresses activating point mutations in K-ras) originally purchased from the American Type Culture Collection (Rockville, MD) were grown in RPMI-1640 with 10% fetal bovine serum without antibiotics. ',\n",
       " '(MDS) was used to represent genetic distances in a two-dimensional space using SPSS ver. 17.0 (SPSS Inc.).',\n",
       " 'An adapted version of Probability Weighted Iterative Generalised Least Squares technique was used for the WAACHS data to account for the complex survey design of children within families within communities using SAS software [33].',\n",
       " 'Automated model building procedures (forward, backward or stepwise regression) are not supported by the survey analysis command in Stata 8.0 and are therefore not used in any of the analyses. ',\n",
       " 'Next, we manipulated voice pitch using the Pitch-Synchronous Overlap Add (PSOLA) algorithm [38] in Praat acoustic phonetics software [39]. ',\n",
       " 'Safety analyses included tabulation of type and frequency of all adverse events.',\n",
       " 'The obtained fMRI data were preprocessed and analyzed using the statistical parametric mapping (SPM8; Wellcome Trust Center for Imaging, London, UK; http://www.fil.ion.ucl.ac.uk/spm). ',\n",
       " 'The software, called MetaSel, is the first attempt to offer a free assistive karyotyping tool for chromosome analysis.\\n',\n",
       " 'The results of a PASS prediction highlight the probability of the activity of these compounds as a caspase-3 stimulant, JAK2 expression inhibitor, and apoptosis agonist, as shown in Table S1. ',\n",
       " 'All statistic analyses were computed using Statistical Package for Social Science (SPSS, Chicago, Illinois, USA) software version 19.0 for Windows. ',\n",
       " 'PyMS is a novel software package for the processing of raw GC-MS data, particularly suitable for scripting of customized processing pipelines and for data processing in batch mode.\\n',\n",
       " 'We used logistic regression with ridge estimators as a classification algorithm [25] as implemented in the Waikato Environment for Knowledge Analysis interface [26].',\n",
       " 'Statistical analysis was carried out using STATA v11 (Texas, USA) unless otherwise stated.\\n\\n',\n",
       " 'Chi-square, t-test, and regression models were constructed in JMP 9.0 (SAS, Inc).\\n\\n',\n",
       " 'The whole mtDNA sequences are available in GenBank under accession numbers MG571104 - MG571221.',\n",
       " 'After mating, the queens developed DWV infections that were highly correlated among different body parts of each queen (head, thorax, abdomen, ovary, spermatheca, and sperm) and differed strongly between individual queens (DWV titers ranging from 0 to >1010) [31]. ',\n",
       " 'All data analyses were performed in SPSS (v.19) unless otherwise stated.\\n\\n',\n",
       " '(iii) Voxel-wise correlations of brain maps with maps from NeuroSynth [33], to find the keywords from the NeuroSynth database whose posterior probability maps are most similar to the participant’s activity patterns. ',\n",
       " 'Calculations were performed with SPSS software (Version 22.2).',\n",
       " 'arm zw3 FRT101/ovoD1 FRT101; aPKC/+ X\\n',\n",
       " 'To apply the transformations to a statistical contrast, the appropriate individual-level contrast (obtained from the GLM as described above) was extracted from the AFNI BRIK file and normalized to template space. ',\n",
       " 'Each subject’s anatomical volume was first aligned to the motion corrected average EPI volume and then to the MNI 152 brain template in AFNI.',\n",
       " '[72], probes not rated as “perfect” (50 out of 50 nucleotides match the reference genome) or “good” (48 to 49 out of 50 matches to the genome) were removed; so were probes that mapped to more than one genomic location according to BLAST queries[73] and probes that included a SNP (NCBI dbSNP Build 137); finally, only probes detected in at least 5% of the mice at a 0.95 detection level (as per GenomeStudio) were retained. ',\n",
       " 'Analyses were conducted with R statistical software V.2.11.1 (http://www.R-project.org).',\n",
       " 'Data cannot be shared publicly because of confidentiality based restrictions placed by the ethics review committee of Aga Khan University Hospital.',\n",
       " 'Data analysis was performed using the SPSS 15.0 PC package (SPSS Inc., Chicago, IL), with statistical significance set at P<0.05.\\n\\n',\n",
       " 'This also restricted interpersonal interaction in 63%.',\n",
       " 'The unphased genotypes of 165 unrelated HapMap3 Caucasians (CEU) were merged by PLINK with the genotypes of the eight participants.',\n",
       " 'It has been widely used to predict species distribution, including tephritid species [4,22] and in addition to its robust statistical properties, MaxEnt shows a high performance across several niche modelling methods for presence-only data [24]. ',\n",
       " 'Prior to the SNP-based linkage analyses in the two families, power analysis with 1,000 simulations was conducted using the SLINK package [59, 60]. ',\n",
       " 'Cluster analysis was used to determine boundaries between diatom assemblage zones in the stratigraphy, with the number of significant zones evaluated by the broken stick model, using the rioja package in R [62, 63]. ',\n",
       " 'The R-Scan module detects thin blood vessels in the conjunctiva and evaluates the degree of redness, using the following scoring scale: 0, no redness; 1, single conjunctival injections, 2, minor, scattered injections; 3, enhanced local injections; and 4, enhanced scattered injections. ',\n",
       " 'In the first step, maximal repeats are found by the software Vmatch (http://www.vmatch.de/), the upgrade of REPuter (22-24).',\n",
       " 'All recordings were acquired using Scan 4.3 (Compumedics Neuroscan) and stored for off-line treatment.',\n",
       " 'Chaste is freely available to download from the website at http://www.cs.ox.ac.uk/chaste.\\n',\n",
       " 'We conducted all statistical analyses in Stata version 13.1 (StataCorp, Texas, USA).\\n\\n',\n",
       " 'Matlab codes are available in figshare along with the raw dataset21.',\n",
       " 'Dimension reduction analysis was performed with t-distributed stochastic neighbor embedding (t-SNE) by Rtsne package in R. Sample correlation analyses were performed on PIVOT [65] or on Cluster (version 3.0) and visualized in TreeView (http://jtreeview.sourceforge.net).',\n",
       " '• Methods and Results: MatrixConverter is an open source program written in Java; a platform-independent binary executable, as well as sample data sets and a user’s manual, are available at https://github.com/gburleigh/MatrixConverter/tree/master/distribution.\\n',\n",
       " 'Peaking calling of the ChIP samples was conducted by comparison with the reads of input using MACS2 [21], with the significance level of p < 0.005. ',\n",
       " 'The ConoMass tool matches peptide masses predicted from transcripts with a list of masses obtained experimentally by proteomics analysis of cone snail venoms.',\n",
       " 'The questionnaire allows for the identification and classification of patients in need of professional psychosocial or psycho-oncologic intervention.',\n",
       " 'The software was developed within the ImageJ environment using a combination of macro script language and available plugins.',\n",
       " 'An interaction between sex and rs1021188 in association with BMDC in the ALSPAC cohort was tested in PLINK, which tests the significance of the interaction term in the linear model. ',\n",
       " 'Here, we provide aLFQ, an open-source implementation of algorithms supporting the estimation of protein quantities by any of the aforementioned methods, and additionally provide automated workflows for data analysis and error estimation.\\n',\n",
       " 'In addition, MIMIC modelling is suitable to validate a construct via reflective modelling [57].\\n\\n',\n",
       " 'For ease of installation (R package dependencies), it is distributed as part of CRAN, http://cran.r-project.org/package=dcGOR.\\n',\n",
       " 'On average, the online survey took approximately 10–15 minutes to complete and data collected was downloaded from Questionpro in SPSS’s data file format. ',\n",
       " 'Calibrations of eye fixations were conducted at the beginning of the experiment using a nine-point fixation procedure as implemented in the EyeLink API (see EyeLink Manual) and using Matlab software.',\n",
       " 'ECG and acceleration data were processed using a program developed on Igor Pro for this purpose (Wavemetrics, USA). ',\n",
       " 'Therefore, an individual can be traced from household to household over the course of his or her life (as long as he or she resides in the Matlab study area). ',\n",
       " 'We used linear mixed models (R library lme4, function lmer) to examine the effects of interactions and to estimate the variance of subtype richness associated with region and selected the best model based on Bayesian Information Criterion (Supplementary Table 3).\\n\\n',\n",
       " \"The VBA toolbox is then used to invert a “dynamical” variant of the Q-learning model (cf. section “Reinforcement learning models of choice data”), given the agent's sequence of choices.\\n\",\n",
       " 'CRISPRFinder core routines were developed in Perl under Debian Linux.',\n",
       " 'For computational fluid dynamics simulations (CFD software 18.0 from Ansys, Canonsburg, USA) three-dimensional models of the microfluidic system with the enclosed larva were created using computer-aided design (CAD software from SolidWorks, Vélizy-Villacoublay, France). ',\n",
       " 'Due to the fairly low statistical threshold of these findings, they should be taken as descriptive and suggestive as opposed to conclusive, but, due to the lack of connectivity data in canines, are still of interest.\\n',\n",
       " 'Mean intensity was calculated using Image J software (NIH) for lines that were 1.5-3.0 microns in length at the dorsal-ventral and anterior-posterior edges of cells in denticle-producing rows 2-5.',\n",
       " 'We used the software “Soscisurvey” (www.soscisurvey.de), which offers all state of the art requirements for online based surveys. ',\n",
       " 'Putative functions of predicted ORFs were manually checked by BLASTp searches against all non-redundant protein sequences from the NCBI database and by functional analysis of deduced protein sequences using InterPro web service (http://www.ebi.ac.uk/interpro/), which allowed protein classification into families and predicted the presence of protein domains. ',\n",
       " 'Association with cancer was calculated using the DAVID web interface (david.abcc. ncifcrf.gov) and as described in ref. 24.',\n",
       " 'This paper presents a thermal convection-based sensor fabricated using simple microelectromechanical systems (MEMS)-based processes.',\n",
       " 'Gene annotation data for these organisms was retrieved directly from MySQL server of BioMart database build 75 (ensembl mart 75; http://www.ensembl.org/info/ data/mysql.html) (25).',\n",
       " 'PyMS currently implements a complete set of methods required in typical GC-MS data processing, including reading of standard data formats, baseline correction, nose filtering, peak detection, peak deconvolution, peak integration, and peak alignment based on dynamic programming described previously [36].\\n',\n",
       " 'To automatize the search for phylogenetically informative markers, we developed a PERL pipeline connected to a MySQL database (available upon request).',\n",
       " 'Maps were created in ArcGIS 9.3 (ESRI, Inc.) and Photofiltre 6.5.1. ',\n",
       " 'No custom code was used to generate or process the data described in the manuscript.',\n",
       " 'In order to investigate whether sex and body weight could influence the inter-individual variation in the behavioural responses (reflected in the PCs), we performed a general linear model (GLM) as well as linear regressions using SAS. ',\n",
       " 'Using ClustalW, we produced a final alignment of 6,468 base pairs per gene.',\n",
       " 'In that case, the phase distribution of the valid areas in both two interferograms will be calculated, as shown in Figure 3d. ',\n",
       " 'Blinding was not necessary since randomization was at the level of HCs.',\n",
       " 'All statistical analyses were conducted using SAS 9.3 software package (SAS Institute, Cary, NC, USA).\\n\\n',\n",
       " 'Here, we present a novel Python-based toolbox called HDDM (hierarchical drift diffusion model), which allows fast and flexible estimation of the the drift-diffusion model and the related linear ballistic accumulator model.\\n',\n",
       " 'Redness of the conjunctiva: the bulbar redness (nasal, lateral, and total) and the limbal degree of redness (nasal, lateral, and total) were automatically and objectively measured and classified with the R-Scan module of the Keratograph 5M. ',\n",
       " 'Table 1 lists clusters with a minimum number of 250 voxels corresponding to 250 mm3.',\n",
       " 'These searchlights are aligned across subjects based on anatomy (movie data) or MSM-All (HCP resting state data); consequently, each locus within a searchlight is similarly aligned across subjects before CHA. ',\n",
       " 'The temperature frequency was converted into degrees centigrade with Dataquest software (Data Sciences International).',\n",
       " 'Data from the tracking software was converted into Microsoft Excel format to allow further data analysis.']"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = dataset['test']\n",
    "test_text = [i['text'] for i in test]\n",
    "test_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "e9cc969c-a6fe-46cc-9c39-f63ca85d116e",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "classifier_list = classifier(test_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {
      "byteLimit": 2048000,
      "rowLimit": 10000
     },
     "inputWidgets": {},
     "nuid": "14da2783-ebdd-4442-a884-be25bc1d672e",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "precision: 0.8722084774881793\n",
      "recall: 0.8796258843337188\n",
      "f1: 0.8754759732466008\n",
      "accuracy: 0.9212410501193318\n"
     ]
    }
   ],
   "source": [
    "true_labels = [i['label'] for i in test]\n",
    "predicted_labels = [label2id[i['label']] for i in classifier_list]\n",
    "\n",
    "precision = precision_score(true_labels, predicted_labels, average='macro')\n",
    "recall = recall_score(true_labels, predicted_labels, average='macro')\n",
    "f1 = f1_score(true_labels, predicted_labels, average='macro')\n",
    "accuracy = accuracy_score(true_labels, predicted_labels)\n",
    "print('precision:', precision)\n",
    "print('recall:', recall)\n",
    "print('f1:', f1)\n",
    "print('accuracy:', accuracy)"
   ]
  }
 ],
 "metadata": {
  "application/vnd.databricks.v1+notebook": {
   "dashboards": [],
   "language": "python",
   "notebookMetadata": {
    "pythonIndentUnit": 4
   },
   "notebookName": "PubmedBERT sentence classification huggingface",
   "widgets": {}
  },
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
